// @aws-sdk/client-athena/dist-cjs/index.js
'use strict';

var middlewareHostHeader = require('@aws-sdk/middleware-host-header');
var middlewareLogger = require('@aws-sdk/middleware-logger');
var middlewareRecursionDetection = require('@aws-sdk/middleware-recursion-detection');
var middlewareUserAgent = require('@aws-sdk/middleware-user-agent');
var configResolver = require('@smithy/config-resolver');
var core = require('@smithy/core');
var schema = require('@smithy/core/schema');
var middlewareContentLength = require('@smithy/middleware-content-length');
var middlewareEndpoint = require('@smithy/middleware-endpoint');
var middlewareRetry = require('@smithy/middleware-retry');
var smithyClient = require('@smithy/smithy-client');
var httpAuthSchemeProvider = require('./auth/httpAuthSchemeProvider');
var runtimeConfig = require('./runtimeConfig');
var regionConfigResolver = require('@aws-sdk/region-config-resolver');
var protocolHttp = require('@smithy/protocol-http');

const resolveClientEndpointParameters = (options) => {
    return Object.assign(options, {
        useDualstackEndpoint: options.useDualstackEndpoint ?? false,
        useFipsEndpoint: options.useFipsEndpoint ?? false,
        defaultSigningName: "athena",
    });
};
const commonParams = {
    UseFIPS: { type: "builtInParams", name: "useFipsEndpoint" },
    Endpoint: { type: "builtInParams", name: "endpoint" },
    Region: { type: "builtInParams", name: "region" },
    UseDualStack: { type: "builtInParams", name: "useDualstackEndpoint" },
};

const getHttpAuthExtensionConfiguration = (runtimeConfig) => {
    const _httpAuthSchemes = runtimeConfig.httpAuthSchemes;
    let _httpAuthSchemeProvider = runtimeConfig.httpAuthSchemeProvider;
    let _credentials = runtimeConfig.credentials;
    return {
        setHttpAuthScheme(httpAuthScheme) {
            const index = _httpAuthSchemes.findIndex((scheme) => scheme.schemeId === httpAuthScheme.schemeId);
            if (index === -1) {
                _httpAuthSchemes.push(httpAuthScheme);
            }
            else {
                _httpAuthSchemes.splice(index, 1, httpAuthScheme);
            }
        },
        httpAuthSchemes() {
            return _httpAuthSchemes;
        },
        setHttpAuthSchemeProvider(httpAuthSchemeProvider) {
            _httpAuthSchemeProvider = httpAuthSchemeProvider;
        },
        httpAuthSchemeProvider() {
            return _httpAuthSchemeProvider;
        },
        setCredentials(credentials) {
            _credentials = credentials;
        },
        credentials() {
            return _credentials;
        },
    };
};
const resolveHttpAuthRuntimeConfig = (config) => {
    return {
        httpAuthSchemes: config.httpAuthSchemes(),
        httpAuthSchemeProvider: config.httpAuthSchemeProvider(),
        credentials: config.credentials(),
    };
};

const resolveRuntimeExtensions = (runtimeConfig, extensions) => {
    const extensionConfiguration = Object.assign(regionConfigResolver.getAwsRegionExtensionConfiguration(runtimeConfig), smithyClient.getDefaultExtensionConfiguration(runtimeConfig), protocolHttp.getHttpHandlerExtensionConfiguration(runtimeConfig), getHttpAuthExtensionConfiguration(runtimeConfig));
    extensions.forEach((extension) => extension.configure(extensionConfiguration));
    return Object.assign(runtimeConfig, regionConfigResolver.resolveAwsRegionExtensionConfiguration(extensionConfiguration), smithyClient.resolveDefaultRuntimeConfig(extensionConfiguration), protocolHttp.resolveHttpHandlerRuntimeConfig(extensionConfiguration), resolveHttpAuthRuntimeConfig(extensionConfiguration));
};

class AthenaClient extends smithyClient.Client {
    config;
    constructor(...[configuration]) {
        const _config_0 = runtimeConfig.getRuntimeConfig(configuration || {});
        super(_config_0);
        this.initConfig = _config_0;
        const _config_1 = resolveClientEndpointParameters(_config_0);
        const _config_2 = middlewareUserAgent.resolveUserAgentConfig(_config_1);
        const _config_3 = middlewareRetry.resolveRetryConfig(_config_2);
        const _config_4 = configResolver.resolveRegionConfig(_config_3);
        const _config_5 = middlewareHostHeader.resolveHostHeaderConfig(_config_4);
        const _config_6 = middlewareEndpoint.resolveEndpointConfig(_config_5);
        const _config_7 = httpAuthSchemeProvider.resolveHttpAuthSchemeConfig(_config_6);
        const _config_8 = resolveRuntimeExtensions(_config_7, configuration?.extensions || []);
        this.config = _config_8;
        this.middlewareStack.use(schema.getSchemaSerdePlugin(this.config));
        this.middlewareStack.use(middlewareUserAgent.getUserAgentPlugin(this.config));
        this.middlewareStack.use(middlewareRetry.getRetryPlugin(this.config));
        this.middlewareStack.use(middlewareContentLength.getContentLengthPlugin(this.config));
        this.middlewareStack.use(middlewareHostHeader.getHostHeaderPlugin(this.config));
        this.middlewareStack.use(middlewareLogger.getLoggerPlugin(this.config));
        this.middlewareStack.use(middlewareRecursionDetection.getRecursionDetectionPlugin(this.config));
        this.middlewareStack.use(core.getHttpAuthSchemeEndpointRuleSetPlugin(this.config, {
            httpAuthSchemeParametersProvider: httpAuthSchemeProvider.defaultAthenaHttpAuthSchemeParametersProvider,
            identityProviderConfigProvider: async (config) => new core.DefaultIdentityProviderConfig({
                "aws.auth#sigv4": config.credentials,
            }),
        }));
        this.middlewareStack.use(core.getHttpSigningPlugin(this.config));
    }
    destroy() {
        super.destroy();
    }
}

class AthenaServiceException extends smithyClient.ServiceException {
    constructor(options) {
        super(options);
        Object.setPrototypeOf(this, AthenaServiceException.prototype);
    }
}

class InternalServerException extends AthenaServiceException {
    name = "InternalServerException";
    $fault = "server";
    Message;
    constructor(opts) {
        super({
            name: "InternalServerException",
            $fault: "server",
            ...opts,
        });
        Object.setPrototypeOf(this, InternalServerException.prototype);
        this.Message = opts.Message;
    }
}
class InvalidRequestException extends AthenaServiceException {
    name = "InvalidRequestException";
    $fault = "client";
    AthenaErrorCode;
    Message;
    constructor(opts) {
        super({
            name: "InvalidRequestException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, InvalidRequestException.prototype);
        this.AthenaErrorCode = opts.AthenaErrorCode;
        this.Message = opts.Message;
    }
}
class TooManyRequestsException extends AthenaServiceException {
    name = "TooManyRequestsException";
    $fault = "client";
    Message;
    Reason;
    constructor(opts) {
        super({
            name: "TooManyRequestsException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, TooManyRequestsException.prototype);
        this.Message = opts.Message;
        this.Reason = opts.Reason;
    }
}
class ResourceNotFoundException extends AthenaServiceException {
    name = "ResourceNotFoundException";
    $fault = "client";
    Message;
    ResourceName;
    constructor(opts) {
        super({
            name: "ResourceNotFoundException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, ResourceNotFoundException.prototype);
        this.Message = opts.Message;
        this.ResourceName = opts.ResourceName;
    }
}
class MetadataException extends AthenaServiceException {
    name = "MetadataException";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "MetadataException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, MetadataException.prototype);
        this.Message = opts.Message;
    }
}
class SessionAlreadyExistsException extends AthenaServiceException {
    name = "SessionAlreadyExistsException";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "SessionAlreadyExistsException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, SessionAlreadyExistsException.prototype);
        this.Message = opts.Message;
    }
}

const _AC = "AclConfiguration";
const _ACd = "AdditionalConfigs";
const _ACdd = "AdditionalConfiguration";
const _AD = "AllocatedDpus";
const _ADPUS = "ApplicationDPUSizes";
const _ADPUSL = "ApplicationDPUSizesList";
const _AE = "AthenaError";
const _AEC = "AthenaErrorCode";
const _ARI = "ApplicationRuntimeId";
const _AT = "AuthToken";
const _ATET = "AuthTokenExpirationTime";
const _ATu = "AuthenticationType";
const _BGNQ = "BatchGetNamedQuery";
const _BGNQI = "BatchGetNamedQueryInput";
const _BGNQO = "BatchGetNamedQueryOutput";
const _BGPS = "BatchGetPreparedStatement";
const _BGPSI = "BatchGetPreparedStatementInput";
const _BGPSO = "BatchGetPreparedStatementOutput";
const _BGQE = "BatchGetQueryExecution";
const _BGQEI = "BatchGetQueryExecutionInput";
const _BGQEO = "BatchGetQueryExecutionOutput";
const _BSCPQ = "BytesScannedCutoffPerQuery";
const _C = "Classification";
const _CA = "CapacityAllocation";
const _CAC = "CapacityAssignmentConfiguration";
const _CAL = "CapacityAssignmentsList";
const _CAa = "CapacityAssignment";
const _CAap = "CapacityAssignments";
const _CB = "CodeBlock";
const _CC = "CalculationConfiguration";
const _CCEC = "CustomerContentEncryptionConfiguration";
const _CCR = "CancelCapacityReservation";
const _CCRI = "CancelCapacityReservationInput";
const _CCRIr = "CreateCapacityReservationInput";
const _CCRO = "CancelCapacityReservationOutput";
const _CCROr = "CreateCapacityReservationOutput";
const _CCRr = "CreateCapacityReservation";
const _CDC = "CreateDataCatalog";
const _CDCI = "CreateDataCatalogInput";
const _CDCO = "CreateDataCatalogOutput";
const _CDS = "CoordinatorDpuSize";
const _CDT = "CompletionDateTime";
const _CEI = "CalculationExecutionId";
const _CI = "ColumnInfo";
const _CIL = "ColumnInfoList";
const _CL = "CalculationsList";
const _CLl = "ClassificationList";
const _CLo = "ColumnList";
const _CN = "CatalogName";
const _CNI = "CreateNotebookInput";
const _CNO = "CreateNotebookOutput";
const _CNQ = "CreateNamedQuery";
const _CNQI = "CreateNamedQueryInput";
const _CNQO = "CreateNamedQueryOutput";
const _CNr = "CreateNotebook";
const _CPNU = "CreatePresignedNotebookUrl";
const _CPNUR = "CreatePresignedNotebookUrlRequest";
const _CPNURr = "CreatePresignedNotebookUrlResponse";
const _CPS = "CreatePreparedStatement";
const _CPSI = "CreatePreparedStatementInput";
const _CPSO = "CreatePreparedStatementOutput";
const _CR = "CalculationResult";
const _CRL = "CapacityReservationsList";
const _CRN = "CapacityReservationName";
const _CRT = "ClientRequestToken";
const _CRa = "CapacityReservation";
const _CRap = "CapacityReservations";
const _CS = "CalculationStatistics";
const _CSa = "CalculationStatus";
const _CSal = "CalculationSummary";
const _CSas = "CaseSensitive";
const _CT = "CreationTime";
const _CTo = "ConnectionType";
const _CTr = "CreateTime";
const _CU = "ConfigurationUpdates";
const _CULP = "CreateUserLevelPrefix";
const _CWG = "CreateWorkGroup";
const _CWGI = "CreateWorkGroupInput";
const _CWGO = "CreateWorkGroupOutput";
const _CWGT = "CopyWorkGroupTags";
const _CWLC = "CloudWatchLoggingConfiguration";
const _Ca = "Calculations";
const _Cat = "Catalog";
const _Ch = "Children";
const _Cl = "Classifications";
const _Co = "Column";
const _Col = "Columns";
const _Com = "Comment";
const _Con = "Configuration";
const _D = "Description";
const _DC = "DataCatalog";
const _DCO = "DeleteCatalogOnly";
const _DCR = "DeleteCapacityReservation";
const _DCRI = "DeleteCapacityReservationInput";
const _DCRO = "DeleteCapacityReservationOutput";
const _DCS = "DataCatalogSummary";
const _DCSL = "DataCatalogSummaryList";
const _DCSa = "DataCatalogsSummary";
const _DCp = "DpuCount";
const _DDC = "DeleteDataCatalog";
const _DDCI = "DeleteDataCatalogInput";
const _DDCO = "DeleteDataCatalogOutput";
const _DEDS = "DefaultExecutorDpuSize";
const _DEIM = "DpuExecutionInMillis";
const _DL = "DatabaseList";
const _DML = "DataManifestLocation";
const _DN = "DatabaseName";
const _DNI = "DeleteNotebookInput";
const _DNO = "DeleteNotebookOutput";
const _DNQ = "DeleteNamedQuery";
const _DNQI = "DeleteNamedQueryInput";
const _DNQO = "DeleteNamedQueryOutput";
const _DNe = "DeleteNotebook";
const _DPS = "DeletePreparedStatement";
const _DPSI = "DeletePreparedStatementInput";
const _DPSO = "DeletePreparedStatementOutput";
const _DSIB = "DataScannedInBytes";
const _DWG = "DeleteWorkGroup";
const _DWGI = "DeleteWorkGroupInput";
const _DWGO = "DeleteWorkGroupOutput";
const _Da = "Database";
const _Dat = "Datum";
const _Data = "Data";
const _E = "Enabled";
const _EBO = "ExpectedBucketOwner";
const _EC = "ErrorCategory";
const _ECn = "EncryptionConfiguration";
const _ECng = "EngineConfiguration";
const _ECr = "ErrorCode";
const _EDT = "EndDateTime";
const _EETIM = "EngineExecutionTimeInMillis";
const _EEV = "EffectiveEngineVersion";
const _EI = "ExecutorId";
const _EIC = "EnableIdentityCenter";
const _EM = "ErrorMessage";
const _EMEC = "EnableMinimumEncryptionConfiguration";
const _EN = "ExportNotebook";
const _ENI = "ExportNotebookInput";
const _ENO = "ExportNotebookOutput";
const _EO = "EncryptionOption";
const _EP = "ExecutionParameters";
const _ER = "ExecutionRole";
const _ES = "ExecutorsSummary";
const _ESAG = "EnableS3AccessGrants";
const _ESF = "ExecutorStateFilter";
const _ESL = "ExecutorsSummaryList";
const _ESx = "ExecutorState";
const _ESxe = "ExecutorSize";
const _ET = "ErrorType";
const _ETx = "ExecutorType";
const _ETxe = "ExecutionTime";
const _EU = "EndpointUrl";
const _EV = "EngineVersion";
const _EVL = "EngineVersionsList";
const _EVn = "EngineVersions";
const _EWGC = "EnforceWorkGroupConfiguration";
const _Er = "Error";
const _Ex = "Expression";
const _F = "Filters";
const _FD = "FilterDefinition";
const _GCAC = "GetCapacityAssignmentConfiguration";
const _GCACI = "GetCapacityAssignmentConfigurationInput";
const _GCACO = "GetCapacityAssignmentConfigurationOutput";
const _GCE = "GetCalculationExecution";
const _GCEC = "GetCalculationExecutionCode";
const _GCECR = "GetCalculationExecutionCodeRequest";
const _GCECRe = "GetCalculationExecutionCodeResponse";
const _GCER = "GetCalculationExecutionRequest";
const _GCERe = "GetCalculationExecutionResponse";
const _GCES = "GetCalculationExecutionStatus";
const _GCESR = "GetCalculationExecutionStatusRequest";
const _GCESRe = "GetCalculationExecutionStatusResponse";
const _GCR = "GetCapacityReservation";
const _GCRI = "GetCapacityReservationInput";
const _GCRO = "GetCapacityReservationOutput";
const _GD = "GetDatabase";
const _GDC = "GetDataCatalog";
const _GDCI = "GetDataCatalogInput";
const _GDCO = "GetDataCatalogOutput";
const _GDI = "GetDatabaseInput";
const _GDO = "GetDatabaseOutput";
const _GNM = "GetNotebookMetadata";
const _GNMI = "GetNotebookMetadataInput";
const _GNMO = "GetNotebookMetadataOutput";
const _GNQ = "GetNamedQuery";
const _GNQI = "GetNamedQueryInput";
const _GNQO = "GetNamedQueryOutput";
const _GPS = "GetPreparedStatement";
const _GPSI = "GetPreparedStatementInput";
const _GPSO = "GetPreparedStatementOutput";
const _GQE = "GetQueryExecution";
const _GQEI = "GetQueryExecutionInput";
const _GQEO = "GetQueryExecutionOutput";
const _GQR = "GetQueryResults";
const _GQRI = "GetQueryResultsInput";
const _GQRO = "GetQueryResultsOutput";
const _GQRS = "GetQueryRuntimeStatistics";
const _GQRSI = "GetQueryRuntimeStatisticsInput";
const _GQRSO = "GetQueryRuntimeStatisticsOutput";
const _GRD = "GetResourceDashboard";
const _GRDR = "GetResourceDashboardRequest";
const _GRDRe = "GetResourceDashboardResponse";
const _GS = "GetSession";
const _GSE = "GetSessionEndpoint";
const _GSER = "GetSessionEndpointRequest";
const _GSERe = "GetSessionEndpointResponse";
const _GSR = "GetSessionRequest";
const _GSRe = "GetSessionResponse";
const _GSS = "GetSessionStatus";
const _GSSR = "GetSessionStatusRequest";
const _GSSRe = "GetSessionStatusResponse";
const _GTM = "GetTableMetadata";
const _GTMI = "GetTableMetadataInput";
const _GTMO = "GetTableMetadataOutput";
const _GWG = "GetWorkGroup";
const _GWGI = "GetWorkGroupInput";
const _GWGO = "GetWorkGroupOutput";
const _I = "Identifier";
const _IB = "InputBytes";
const _ICAA = "IdentityCenterApplicationArn";
const _ICC = "IdentityCenterConfiguration";
const _ICIA = "IdentityCenterInstanceArn";
const _IN = "ImportNotebook";
const _INI = "ImportNotebookInput";
const _INO = "ImportNotebookOutput";
const _IR = "InputRows";
const _IRE = "InvalidRequestException";
const _ISDT = "IdleSinceDateTime";
const _ISE = "InternalServerException";
const _ITS = "IdleTimeoutSeconds";
const _K = "Key";
const _KK = "KmsKey";
const _L = "Label";
const _LA = "LastAllocation";
const _LADPUS = "ListApplicationDPUSizes";
const _LADPUSI = "ListApplicationDPUSizesInput";
const _LADPUSO = "ListApplicationDPUSizesOutput";
const _LAT = "LastAccessTime";
const _LCE = "ListCalculationExecutions";
const _LCER = "ListCalculationExecutionsRequest";
const _LCERi = "ListCalculationExecutionsResponse";
const _LCR = "ListCapacityReservations";
const _LCRI = "ListCapacityReservationsInput";
const _LCRO = "ListCapacityReservationsOutput";
const _LD = "ListDatabases";
const _LDC = "ListDataCatalogs";
const _LDCI = "ListDataCatalogsInput";
const _LDCO = "ListDataCatalogsOutput";
const _LDI = "ListDatabasesInput";
const _LDO = "ListDatabasesOutput";
const _LE = "ListExecutors";
const _LER = "ListExecutorsRequest";
const _LERi = "ListExecutorsResponse";
const _LEV = "ListEngineVersions";
const _LEVI = "ListEngineVersionsInput";
const _LEVO = "ListEngineVersionsOutput";
const _LG = "LogGroup";
const _LL = "LogLocation";
const _LMDT = "LastModifiedDateTime";
const _LMT = "LastModifiedTime";
const _LNM = "ListNotebookMetadata";
const _LNMI = "ListNotebookMetadataInput";
const _LNMO = "ListNotebookMetadataOutput";
const _LNQ = "ListNamedQueries";
const _LNQI = "ListNamedQueriesInput";
const _LNQO = "ListNamedQueriesOutput";
const _LNS = "ListNotebookSessions";
const _LNSR = "ListNotebookSessionsRequest";
const _LNSRi = "ListNotebookSessionsResponse";
const _LPS = "ListPreparedStatements";
const _LPSI = "ListPreparedStatementsInput";
const _LPSO = "ListPreparedStatementsOutput";
const _LQE = "ListQueryExecutions";
const _LQEI = "ListQueryExecutionsInput";
const _LQEO = "ListQueryExecutionsOutput";
const _LS = "ListSessions";
const _LSAT = "LastSuccessfulAllocationTime";
const _LSNP = "LogStreamNamePrefix";
const _LSR = "ListSessionsRequest";
const _LSRi = "ListSessionsResponse";
const _LT = "LogTypes";
const _LTFR = "ListTagsForResource";
const _LTFRI = "ListTagsForResourceInput";
const _LTFRO = "ListTagsForResourceOutput";
const _LTM = "LogTypesMap";
const _LTMI = "ListTableMetadataInput";
const _LTMO = "ListTableMetadataOutput";
const _LTMi = "ListTableMetadata";
const _LWG = "ListWorkGroups";
const _LWGI = "ListWorkGroupsInput";
const _LWGO = "ListWorkGroupsOutput";
const _M = "Message";
const _MAIM = "MaxAgeInMinutes";
const _MC = "MonitoringConfiguration";
const _MCD = "MaxConcurrentDpus";
const _ME = "MetadataException";
const _MLC = "ManagedLoggingConfiguration";
const _MQRC = "ManagedQueryResultsConfiguration";
const _MQRCU = "ManagedQueryResultsConfigurationUpdates";
const _MQREC = "ManagedQueryResultsEncryptionConfiguration";
const _MR = "MaxResults";
const _N = "Name";
const _NI = "NotebookId";
const _NM = "NotebookMetadata";
const _NMA = "NotebookMetadataArray";
const _NML = "NotebookMetadataList";
const _NQ = "NamedQueries";
const _NQI = "NamedQueryIds";
const _NQIa = "NamedQueryId";
const _NQL = "NamedQueryList";
const _NQa = "NamedQuery";
const _NSL = "NotebookSessionsList";
const _NSLU = "NotebookS3LocationUri";
const _NSS = "NotebookSessionSummary";
const _NT = "NextToken";
const _NU = "NotebookUrl";
const _NV = "NotebookVersion";
const _Nu = "Nullable";
const _OB = "OutputBytes";
const _OL = "OutputLocation";
const _OR = "OutputRows";
const _OS = "OutputStage";
const _P = "Progress";
const _PCAC = "PutCapacityAssignmentConfiguration";
const _PCACI = "PutCapacityAssignmentConfigurationInput";
const _PCACO = "PutCapacityAssignmentConfigurationOutput";
const _PCWME = "PublishCloudWatchMetricsEnabled";
const _PK = "PartitionKeys";
const _PS = "PreparedStatements";
const _PSDL = "PreparedStatementDetailsList";
const _PSL = "PreparedStatementsList";
const _PSN = "PreparedStatementNames";
const _PSS = "PreparedStatementSummary";
const _PSr = "PreparedStatement";
const _Pa = "Parameters";
const _Pay = "Payload";
const _Pr = "Properties";
const _Pre = "Precision";
const _Q = "Query";
const _QE = "QueryExecutions";
const _QEC = "QueryExecutionContext";
const _QEI = "QueryExecutionIds";
const _QEIu = "QueryExecutionId";
const _QEL = "QueryExecutionList";
const _QES = "QueryExecutionStatistics";
const _QESu = "QueryExecutionStatus";
const _QEu = "QueryExecution";
const _QPTIM = "QueryPlanningTimeInMillis";
const _QQTIM = "QueryQueueTimeInMillis";
const _QRS = "QueryRuntimeStatistics";
const _QRSAGC = "QueryResultsS3AccessGrantsConfiguration";
const _QRSR = "QueryRuntimeStatisticsRows";
const _QRST = "QueryRuntimeStatisticsTimeline";
const _QRT = "QueryResultType";
const _QS = "QueryString";
const _QSP = "QueryStagePlan";
const _QSPN = "QueryStagePlanNode";
const _QSPNu = "QueryStagePlanNodes";
const _QSu = "QueryStatement";
const _QSue = "QueryStage";
const _QSuer = "QueryStages";
const _R = "Retryable";
const _RAC = "RemoveAclConfiguration";
const _RARN = "ResourceARN";
const _RBSCPQ = "RemoveBytesScannedCutoffPerQuery";
const _RC = "ResultConfiguration";
const _RCCEC = "RemoveCustomerContentEncryptionConfiguration";
const _RCT = "RequestCompletionTime";
const _RCU = "ResultConfigurationUpdates";
const _RDO = "RecursiveDeleteOption";
const _REBO = "RemoveExpectedBucketOwner";
const _REC = "RemoveEncryptionConfiguration";
const _RL = "RowList";
const _RN = "ResourceName";
const _RNFE = "ResourceNotFoundException";
const _ROL = "RemoveOutputLocation";
const _RPE = "RequesterPaysEnabled";
const _RPR = "ReusedPreviousResult";
const _RRBAC = "ResultReuseByAgeConfiguration";
const _RRC = "ResultReuseConfiguration";
const _RRI = "ResultReuseInformation";
const _RS = "ResultSet";
const _RSM = "ResultSetMetadata";
const _RSU = "ResultS3Uri";
const _RSe = "RemoteSources";
const _RT = "ResultType";
const _RTe = "RequestTime";
const _Re = "Result";
const _Rea = "Reason";
const _Ro = "Rows";
const _Row = "Row";
const _S = "State";
const _SAEE = "SessionAlreadyExistsException";
const _SAO = "S3AclOption";
const _SC = "SessionConfiguration";
const _SCE = "StartCalculationExecution";
const _SCER = "StartCalculationExecutionRequest";
const _SCERt = "StartCalculationExecutionResponse";
const _SCERto = "StopCalculationExecutionRequest";
const _SCERtop = "StopCalculationExecutionResponse";
const _SCEt = "StopCalculationExecution";
const _SCR = "StateChangeReason";
const _SDPUS = "SupportedDPUSizes";
const _SDT = "SubmissionDateTime";
const _SDTt = "StartDateTime";
const _SESU = "StdErrorS3Uri";
const _SEV = "SelectedEngineVersion";
const _SF = "StateFilter";
const _SI = "SessionId";
const _SITIM = "SessionIdleTimeoutInMinutes";
const _SIt = "StageId";
const _SL = "SessionsList";
const _SLC = "S3LoggingConfiguration";
const _SM = "StatusMessage";
const _SN = "SchemaName";
const _SNt = "StatementName";
const _SOSU = "StdOutS3Uri";
const _SP = "SparkProperties";
const _SPPTIM = "ServicePreProcessingTimeInMillis";
const _SPTIM = "ServiceProcessingTimeInMillis";
const _SQE = "StartQueryExecution";
const _SQEI = "StartQueryExecutionInput";
const _SQEIt = "StopQueryExecutionInput";
const _SQEO = "StartQueryExecutionOutput";
const _SQEOt = "StopQueryExecutionOutput";
const _SQEt = "StopQueryExecution";
const _SS = "SubStages";
const _SSR = "StartSessionRequest";
const _SSRt = "StartSessionResponse";
const _SSe = "SessionStatistics";
const _SSes = "SessionStatus";
const _SSess = "SessionSummary";
const _SSt = "StartSession";
const _ST = "StatementType";
const _STu = "SubstatementType";
const _Sc = "Scale";
const _Se = "Sessions";
const _St = "Status";
const _Sta = "Statistics";
const _T = "Type";
const _TD = "TargetDpus";
const _TDT = "TerminationDateTime";
const _TETIM = "TotalExecutionTimeInMillis";
const _TK = "TagKeys";
const _TL = "TagList";
const _TM = "TableMetadata";
const _TML = "TableMetadataList";
const _TMRE = "TooManyRequestsException";
const _TN = "TableName";
const _TR = "TagResource";
const _TRI = "TagResourceInput";
const _TRO = "TagResourceOutput";
const _TS = "TerminateSession";
const _TSR = "TerminateSessionRequest";
const _TSRe = "TerminateSessionResponse";
const _TT = "TableType";
const _Ta = "Tags";
const _Tag = "Tag";
const _Ti = "Timeline";
const _U = "Url";
const _UC = "UpdateCount";
const _UCR = "UpdateCapacityReservation";
const _UCRI = "UpdateCapacityReservationInput";
const _UCRO = "UpdateCapacityReservationOutput";
const _UDC = "UpdateDataCatalog";
const _UDCI = "UpdateDataCatalogInput";
const _UDCO = "UpdateDataCatalogOutput";
const _UN = "UpdateNotebook";
const _UNI = "UpdateNotebookInput";
const _UNM = "UpdateNotebookMetadata";
const _UNMI = "UpdateNotebookMetadataInput";
const _UNMO = "UpdateNotebookMetadataOutput";
const _UNO = "UpdateNotebookOutput";
const _UNQ = "UpdateNamedQuery";
const _UNQI = "UnprocessedNamedQueryIds";
const _UNQIL = "UnprocessedNamedQueryIdList";
const _UNQIn = "UnprocessedNamedQueryId";
const _UNQIp = "UpdateNamedQueryInput";
const _UNQO = "UpdateNamedQueryOutput";
const _UPS = "UpdatePreparedStatement";
const _UPSI = "UpdatePreparedStatementInput";
const _UPSN = "UnprocessedPreparedStatementNames";
const _UPSNL = "UnprocessedPreparedStatementNameList";
const _UPSNn = "UnprocessedPreparedStatementName";
const _UPSO = "UpdatePreparedStatementOutput";
const _UQEI = "UnprocessedQueryExecutionIds";
const _UQEIL = "UnprocessedQueryExecutionIdList";
const _UQEIn = "UnprocessedQueryExecutionId";
const _UR = "UntagResource";
const _URI = "UntagResourceInput";
const _URO = "UntagResourceOutput";
const _UWG = "UpdateWorkGroup";
const _UWGI = "UpdateWorkGroupInput";
const _UWGO = "UpdateWorkGroupOutput";
const _V = "Value";
const _VCV = "VarCharValue";
const _WD = "WorkingDirectory";
const _WG = "WorkGroup";
const _WGC = "WorkGroupConfiguration";
const _WGCU = "WorkGroupConfigurationUpdates";
const _WGL = "WorkGroupsList";
const _WGN = "WorkGroupNames";
const _WGNo = "WorkGroupName";
const _WGS = "WorkGroupSummary";
const _WGo = "WorkGroups";
const _c = "client";
const _dL = "datumList";
const _e = "error";
const _s = "server";
const _sm = "smithy.ts.sdk.synthetic.com.amazonaws.athena";
const n0 = "com.amazonaws.athena";
var AclConfiguration$ = [3, n0, _AC,
    0,
    [_SAO],
    [0], 1
];
var ApplicationDPUSizes$ = [3, n0, _ADPUS,
    0,
    [_ARI, _SDPUS],
    [0, 64 | 1]
];
var AthenaError$ = [3, n0, _AE,
    0,
    [_EC, _ET, _R, _EM],
    [1, 1, 2, 0]
];
var BatchGetNamedQueryInput$ = [3, n0, _BGNQI,
    0,
    [_NQI],
    [64 | 0], 1
];
var BatchGetNamedQueryOutput$ = [3, n0, _BGNQO,
    0,
    [_NQ, _UNQI],
    [() => NamedQueryList, () => UnprocessedNamedQueryIdList]
];
var BatchGetPreparedStatementInput$ = [3, n0, _BGPSI,
    0,
    [_PSN, _WG],
    [64 | 0, 0], 2
];
var BatchGetPreparedStatementOutput$ = [3, n0, _BGPSO,
    0,
    [_PS, _UPSN],
    [() => PreparedStatementDetailsList, () => UnprocessedPreparedStatementNameList]
];
var BatchGetQueryExecutionInput$ = [3, n0, _BGQEI,
    0,
    [_QEI],
    [64 | 0], 1
];
var BatchGetQueryExecutionOutput$ = [3, n0, _BGQEO,
    0,
    [_QE, _UQEI],
    [() => QueryExecutionList, () => UnprocessedQueryExecutionIdList]
];
var CalculationConfiguration$ = [3, n0, _CC,
    0,
    [_CB],
    [0]
];
var CalculationResult$ = [3, n0, _CR,
    0,
    [_SOSU, _SESU, _RSU, _RT],
    [0, 0, 0, 0]
];
var CalculationStatistics$ = [3, n0, _CS,
    0,
    [_DEIM, _P],
    [1, 0]
];
var CalculationStatus$ = [3, n0, _CSa,
    0,
    [_SDT, _CDT, _S, _SCR],
    [4, 4, 0, 0]
];
var CalculationSummary$ = [3, n0, _CSal,
    0,
    [_CEI, _D, _St],
    [0, 0, () => CalculationStatus$]
];
var CancelCapacityReservationInput$ = [3, n0, _CCRI,
    0,
    [_N],
    [0], 1
];
var CancelCapacityReservationOutput$ = [3, n0, _CCRO,
    0,
    [],
    []
];
var CapacityAllocation$ = [3, n0, _CA,
    0,
    [_St, _RTe, _SM, _RCT],
    [0, 4, 0, 4], 2
];
var CapacityAssignment$ = [3, n0, _CAa,
    0,
    [_WGN],
    [64 | 0]
];
var CapacityAssignmentConfiguration$ = [3, n0, _CAC,
    0,
    [_CRN, _CAap],
    [0, () => CapacityAssignmentsList]
];
var CapacityReservation$ = [3, n0, _CRa,
    0,
    [_N, _St, _TD, _AD, _CT, _LA, _LSAT],
    [0, 0, 1, 1, 4, () => CapacityAllocation$, 4], 5
];
var Classification$ = [3, n0, _C,
    0,
    [_N, _Pr],
    [0, 128 | 0]
];
var CloudWatchLoggingConfiguration$ = [3, n0, _CWLC,
    0,
    [_E, _LG, _LSNP, _LT],
    [2, 0, 0, [2, n0, _LTM, 0, 0, 64 | 0]], 1
];
var Column$ = [3, n0, _Co,
    0,
    [_N, _T, _Com],
    [0, 0, 0], 1
];
var ColumnInfo$ = [3, n0, _CI,
    0,
    [_N, _T, _CN, _SN, _TN, _L, _Pre, _Sc, _Nu, _CSas],
    [0, 0, 0, 0, 0, 0, 1, 1, 0, 2], 2
];
var CreateCapacityReservationInput$ = [3, n0, _CCRIr,
    0,
    [_TD, _N, _Ta],
    [1, 0, () => TagList], 2
];
var CreateCapacityReservationOutput$ = [3, n0, _CCROr,
    0,
    [],
    []
];
var CreateDataCatalogInput$ = [3, n0, _CDCI,
    0,
    [_N, _T, _D, _Pa, _Ta],
    [0, 0, 0, 128 | 0, () => TagList], 2
];
var CreateDataCatalogOutput$ = [3, n0, _CDCO,
    0,
    [_DC],
    [() => DataCatalog$]
];
var CreateNamedQueryInput$ = [3, n0, _CNQI,
    0,
    [_N, _Da, _QS, _D, _CRT, _WG],
    [0, 0, 0, 0, [0, 4], 0], 3
];
var CreateNamedQueryOutput$ = [3, n0, _CNQO,
    0,
    [_NQIa],
    [0]
];
var CreateNotebookInput$ = [3, n0, _CNI,
    0,
    [_WG, _N, _CRT],
    [0, 0, 0], 2
];
var CreateNotebookOutput$ = [3, n0, _CNO,
    0,
    [_NI],
    [0]
];
var CreatePreparedStatementInput$ = [3, n0, _CPSI,
    0,
    [_SNt, _WG, _QSu, _D],
    [0, 0, 0, 0], 3
];
var CreatePreparedStatementOutput$ = [3, n0, _CPSO,
    0,
    [],
    []
];
var CreatePresignedNotebookUrlRequest$ = [3, n0, _CPNUR,
    0,
    [_SI],
    [0], 1
];
var CreatePresignedNotebookUrlResponse$ = [3, n0, _CPNURr,
    0,
    [_NU, _AT, _ATET],
    [0, 0, 1], 3
];
var CreateWorkGroupInput$ = [3, n0, _CWGI,
    0,
    [_N, _Con, _D, _Ta],
    [0, () => WorkGroupConfiguration$, 0, () => TagList], 1
];
var CreateWorkGroupOutput$ = [3, n0, _CWGO,
    0,
    [],
    []
];
var CustomerContentEncryptionConfiguration$ = [3, n0, _CCEC,
    0,
    [_KK],
    [0], 1
];
var Database$ = [3, n0, _Da,
    0,
    [_N, _D, _Pa],
    [0, 0, 128 | 0], 1
];
var DataCatalog$ = [3, n0, _DC,
    0,
    [_N, _T, _D, _Pa, _St, _CTo, _Er],
    [0, 0, 0, 128 | 0, 0, 0, 0], 2
];
var DataCatalogSummary$ = [3, n0, _DCS,
    0,
    [_CN, _T, _St, _CTo, _Er],
    [0, 0, 0, 0, 0]
];
var Datum$ = [3, n0, _Dat,
    0,
    [_VCV],
    [0]
];
var DeleteCapacityReservationInput$ = [3, n0, _DCRI,
    0,
    [_N],
    [0], 1
];
var DeleteCapacityReservationOutput$ = [3, n0, _DCRO,
    0,
    [],
    []
];
var DeleteDataCatalogInput$ = [3, n0, _DDCI,
    0,
    [_N, _DCO],
    [0, 2], 1
];
var DeleteDataCatalogOutput$ = [3, n0, _DDCO,
    0,
    [_DC],
    [() => DataCatalog$]
];
var DeleteNamedQueryInput$ = [3, n0, _DNQI,
    0,
    [_NQIa],
    [[0, 4]]
];
var DeleteNamedQueryOutput$ = [3, n0, _DNQO,
    0,
    [],
    []
];
var DeleteNotebookInput$ = [3, n0, _DNI,
    0,
    [_NI],
    [0], 1
];
var DeleteNotebookOutput$ = [3, n0, _DNO,
    0,
    [],
    []
];
var DeletePreparedStatementInput$ = [3, n0, _DPSI,
    0,
    [_SNt, _WG],
    [0, 0], 2
];
var DeletePreparedStatementOutput$ = [3, n0, _DPSO,
    0,
    [],
    []
];
var DeleteWorkGroupInput$ = [3, n0, _DWGI,
    0,
    [_WG, _RDO],
    [0, 2], 1
];
var DeleteWorkGroupOutput$ = [3, n0, _DWGO,
    0,
    [],
    []
];
var EncryptionConfiguration$ = [3, n0, _ECn,
    0,
    [_EO, _KK],
    [0, 0], 1
];
var EngineConfiguration$ = [3, n0, _ECng,
    0,
    [_CDS, _MCD, _DEDS, _ACd, _SP, _Cl],
    [1, 1, 1, 128 | 0, 128 | 0, () => ClassificationList]
];
var EngineVersion$ = [3, n0, _EV,
    0,
    [_SEV, _EEV],
    [0, 0]
];
var ExecutorsSummary$ = [3, n0, _ES,
    0,
    [_EI, _ETx, _SDTt, _TDT, _ESx, _ESxe],
    [0, 0, 1, 1, 0, 1], 1
];
var ExportNotebookInput$ = [3, n0, _ENI,
    0,
    [_NI],
    [0], 1
];
var ExportNotebookOutput$ = [3, n0, _ENO,
    0,
    [_NM, _Pay],
    [() => NotebookMetadata$, 0]
];
var FilterDefinition$ = [3, n0, _FD,
    0,
    [_N],
    [0]
];
var GetCalculationExecutionCodeRequest$ = [3, n0, _GCECR,
    0,
    [_CEI],
    [0], 1
];
var GetCalculationExecutionCodeResponse$ = [3, n0, _GCECRe,
    0,
    [_CB],
    [0]
];
var GetCalculationExecutionRequest$ = [3, n0, _GCER,
    0,
    [_CEI],
    [0], 1
];
var GetCalculationExecutionResponse$ = [3, n0, _GCERe,
    0,
    [_CEI, _SI, _D, _WD, _St, _Sta, _Re],
    [0, 0, 0, 0, () => CalculationStatus$, () => CalculationStatistics$, () => CalculationResult$]
];
var GetCalculationExecutionStatusRequest$ = [3, n0, _GCESR,
    0,
    [_CEI],
    [0], 1
];
var GetCalculationExecutionStatusResponse$ = [3, n0, _GCESRe,
    0,
    [_St, _Sta],
    [() => CalculationStatus$, () => CalculationStatistics$]
];
var GetCapacityAssignmentConfigurationInput$ = [3, n0, _GCACI,
    0,
    [_CRN],
    [0], 1
];
var GetCapacityAssignmentConfigurationOutput$ = [3, n0, _GCACO,
    0,
    [_CAC],
    [() => CapacityAssignmentConfiguration$], 1
];
var GetCapacityReservationInput$ = [3, n0, _GCRI,
    0,
    [_N],
    [0], 1
];
var GetCapacityReservationOutput$ = [3, n0, _GCRO,
    0,
    [_CRa],
    [() => CapacityReservation$], 1
];
var GetDatabaseInput$ = [3, n0, _GDI,
    0,
    [_CN, _DN, _WG],
    [0, 0, 0], 2
];
var GetDatabaseOutput$ = [3, n0, _GDO,
    0,
    [_Da],
    [() => Database$]
];
var GetDataCatalogInput$ = [3, n0, _GDCI,
    0,
    [_N, _WG],
    [0, 0], 1
];
var GetDataCatalogOutput$ = [3, n0, _GDCO,
    0,
    [_DC],
    [() => DataCatalog$]
];
var GetNamedQueryInput$ = [3, n0, _GNQI,
    0,
    [_NQIa],
    [0], 1
];
var GetNamedQueryOutput$ = [3, n0, _GNQO,
    0,
    [_NQa],
    [() => NamedQuery$]
];
var GetNotebookMetadataInput$ = [3, n0, _GNMI,
    0,
    [_NI],
    [0], 1
];
var GetNotebookMetadataOutput$ = [3, n0, _GNMO,
    0,
    [_NM],
    [() => NotebookMetadata$]
];
var GetPreparedStatementInput$ = [3, n0, _GPSI,
    0,
    [_SNt, _WG],
    [0, 0], 2
];
var GetPreparedStatementOutput$ = [3, n0, _GPSO,
    0,
    [_PSr],
    [() => PreparedStatement$]
];
var GetQueryExecutionInput$ = [3, n0, _GQEI,
    0,
    [_QEIu],
    [0], 1
];
var GetQueryExecutionOutput$ = [3, n0, _GQEO,
    0,
    [_QEu],
    [() => QueryExecution$]
];
var GetQueryResultsInput$ = [3, n0, _GQRI,
    0,
    [_QEIu, _NT, _MR, _QRT],
    [0, 0, 1, 0], 1
];
var GetQueryResultsOutput$ = [3, n0, _GQRO,
    0,
    [_UC, _RS, _NT],
    [1, () => ResultSet$, 0]
];
var GetQueryRuntimeStatisticsInput$ = [3, n0, _GQRSI,
    0,
    [_QEIu],
    [0], 1
];
var GetQueryRuntimeStatisticsOutput$ = [3, n0, _GQRSO,
    0,
    [_QRS],
    [() => QueryRuntimeStatistics$]
];
var GetResourceDashboardRequest$ = [3, n0, _GRDR,
    0,
    [_RARN],
    [0], 1
];
var GetResourceDashboardResponse$ = [3, n0, _GRDRe,
    0,
    [_U],
    [0], 1
];
var GetSessionEndpointRequest$ = [3, n0, _GSER,
    0,
    [_SI],
    [0], 1
];
var GetSessionEndpointResponse$ = [3, n0, _GSERe,
    0,
    [_EU, _AT, _ATET],
    [0, 0, 4], 3
];
var GetSessionRequest$ = [3, n0, _GSR,
    0,
    [_SI],
    [0], 1
];
var GetSessionResponse$ = [3, n0, _GSRe,
    0,
    [_SI, _D, _WG, _EV, _ECng, _NV, _MC, _SC, _St, _Sta],
    [0, 0, 0, 0, () => EngineConfiguration$, 0, () => MonitoringConfiguration$, () => SessionConfiguration$, () => SessionStatus$, () => SessionStatistics$]
];
var GetSessionStatusRequest$ = [3, n0, _GSSR,
    0,
    [_SI],
    [0], 1
];
var GetSessionStatusResponse$ = [3, n0, _GSSRe,
    0,
    [_SI, _St],
    [0, () => SessionStatus$]
];
var GetTableMetadataInput$ = [3, n0, _GTMI,
    0,
    [_CN, _DN, _TN, _WG],
    [0, 0, 0, 0], 3
];
var GetTableMetadataOutput$ = [3, n0, _GTMO,
    0,
    [_TM],
    [() => TableMetadata$]
];
var GetWorkGroupInput$ = [3, n0, _GWGI,
    0,
    [_WG],
    [0], 1
];
var GetWorkGroupOutput$ = [3, n0, _GWGO,
    0,
    [_WG],
    [() => WorkGroup$]
];
var IdentityCenterConfiguration$ = [3, n0, _ICC,
    0,
    [_EIC, _ICIA],
    [2, 0]
];
var ImportNotebookInput$ = [3, n0, _INI,
    0,
    [_WG, _N, _T, _Pay, _NSLU, _CRT],
    [0, 0, 0, 0, 0, 0], 3
];
var ImportNotebookOutput$ = [3, n0, _INO,
    0,
    [_NI],
    [0]
];
var InternalServerException$ = [-3, n0, _ISE,
    { [_e]: _s },
    [_M],
    [0]
];
schema.TypeRegistry.for(n0).registerError(InternalServerException$, InternalServerException);
var InvalidRequestException$ = [-3, n0, _IRE,
    { [_e]: _c },
    [_AEC, _M],
    [0, 0]
];
schema.TypeRegistry.for(n0).registerError(InvalidRequestException$, InvalidRequestException);
var ListApplicationDPUSizesInput$ = [3, n0, _LADPUSI,
    0,
    [_MR, _NT],
    [1, 0]
];
var ListApplicationDPUSizesOutput$ = [3, n0, _LADPUSO,
    0,
    [_ADPUS, _NT],
    [() => ApplicationDPUSizesList, 0]
];
var ListCalculationExecutionsRequest$ = [3, n0, _LCER,
    0,
    [_SI, _SF, _MR, _NT],
    [0, 0, 1, 0], 1
];
var ListCalculationExecutionsResponse$ = [3, n0, _LCERi,
    0,
    [_NT, _Ca],
    [0, () => CalculationsList]
];
var ListCapacityReservationsInput$ = [3, n0, _LCRI,
    0,
    [_NT, _MR],
    [0, 1]
];
var ListCapacityReservationsOutput$ = [3, n0, _LCRO,
    0,
    [_CRap, _NT],
    [() => CapacityReservationsList, 0], 1
];
var ListDatabasesInput$ = [3, n0, _LDI,
    0,
    [_CN, _NT, _MR, _WG],
    [0, 0, 1, 0], 1
];
var ListDatabasesOutput$ = [3, n0, _LDO,
    0,
    [_DL, _NT],
    [() => DatabaseList, 0]
];
var ListDataCatalogsInput$ = [3, n0, _LDCI,
    0,
    [_NT, _MR, _WG],
    [0, 1, 0]
];
var ListDataCatalogsOutput$ = [3, n0, _LDCO,
    0,
    [_DCSa, _NT],
    [() => DataCatalogSummaryList, 0]
];
var ListEngineVersionsInput$ = [3, n0, _LEVI,
    0,
    [_NT, _MR],
    [0, 1]
];
var ListEngineVersionsOutput$ = [3, n0, _LEVO,
    0,
    [_EVn, _NT],
    [() => EngineVersionsList, 0]
];
var ListExecutorsRequest$ = [3, n0, _LER,
    0,
    [_SI, _ESF, _MR, _NT],
    [0, 0, 1, 0], 1
];
var ListExecutorsResponse$ = [3, n0, _LERi,
    0,
    [_SI, _NT, _ES],
    [0, 0, () => ExecutorsSummaryList], 1
];
var ListNamedQueriesInput$ = [3, n0, _LNQI,
    0,
    [_NT, _MR, _WG],
    [0, 1, 0]
];
var ListNamedQueriesOutput$ = [3, n0, _LNQO,
    0,
    [_NQI, _NT],
    [64 | 0, 0]
];
var ListNotebookMetadataInput$ = [3, n0, _LNMI,
    0,
    [_WG, _F, _NT, _MR],
    [0, () => FilterDefinition$, 0, 1], 1
];
var ListNotebookMetadataOutput$ = [3, n0, _LNMO,
    0,
    [_NT, _NML],
    [0, () => NotebookMetadataArray]
];
var ListNotebookSessionsRequest$ = [3, n0, _LNSR,
    0,
    [_NI, _MR, _NT],
    [0, 1, 0], 1
];
var ListNotebookSessionsResponse$ = [3, n0, _LNSRi,
    0,
    [_NSL, _NT],
    [() => NotebookSessionsList, 0], 1
];
var ListPreparedStatementsInput$ = [3, n0, _LPSI,
    0,
    [_WG, _NT, _MR],
    [0, 0, 1], 1
];
var ListPreparedStatementsOutput$ = [3, n0, _LPSO,
    0,
    [_PS, _NT],
    [() => PreparedStatementsList, 0]
];
var ListQueryExecutionsInput$ = [3, n0, _LQEI,
    0,
    [_NT, _MR, _WG],
    [0, 1, 0]
];
var ListQueryExecutionsOutput$ = [3, n0, _LQEO,
    0,
    [_QEI, _NT],
    [64 | 0, 0]
];
var ListSessionsRequest$ = [3, n0, _LSR,
    0,
    [_WG, _SF, _MR, _NT],
    [0, 0, 1, 0], 1
];
var ListSessionsResponse$ = [3, n0, _LSRi,
    0,
    [_NT, _Se],
    [0, () => SessionsList]
];
var ListTableMetadataInput$ = [3, n0, _LTMI,
    0,
    [_CN, _DN, _Ex, _NT, _MR, _WG],
    [0, 0, 0, 0, 1, 0], 2
];
var ListTableMetadataOutput$ = [3, n0, _LTMO,
    0,
    [_TML, _NT],
    [() => TableMetadataList, 0]
];
var ListTagsForResourceInput$ = [3, n0, _LTFRI,
    0,
    [_RARN, _NT, _MR],
    [0, 0, 1], 1
];
var ListTagsForResourceOutput$ = [3, n0, _LTFRO,
    0,
    [_Ta, _NT],
    [() => TagList, 0]
];
var ListWorkGroupsInput$ = [3, n0, _LWGI,
    0,
    [_NT, _MR],
    [0, 1]
];
var ListWorkGroupsOutput$ = [3, n0, _LWGO,
    0,
    [_WGo, _NT],
    [() => WorkGroupsList, 0]
];
var ManagedLoggingConfiguration$ = [3, n0, _MLC,
    0,
    [_E, _KK],
    [2, 0], 1
];
var ManagedQueryResultsConfiguration$ = [3, n0, _MQRC,
    0,
    [_E, _ECn],
    [2, () => ManagedQueryResultsEncryptionConfiguration$], 1
];
var ManagedQueryResultsConfigurationUpdates$ = [3, n0, _MQRCU,
    0,
    [_E, _ECn, _REC],
    [2, () => ManagedQueryResultsEncryptionConfiguration$, 2]
];
var ManagedQueryResultsEncryptionConfiguration$ = [3, n0, _MQREC,
    0,
    [_KK],
    [0], 1
];
var MetadataException$ = [-3, n0, _ME,
    { [_e]: _c },
    [_M],
    [0]
];
schema.TypeRegistry.for(n0).registerError(MetadataException$, MetadataException);
var MonitoringConfiguration$ = [3, n0, _MC,
    0,
    [_CWLC, _MLC, _SLC],
    [() => CloudWatchLoggingConfiguration$, () => ManagedLoggingConfiguration$, () => S3LoggingConfiguration$]
];
var NamedQuery$ = [3, n0, _NQa,
    0,
    [_N, _Da, _QS, _D, _NQIa, _WG],
    [0, 0, 0, 0, 0, 0], 3
];
var NotebookMetadata$ = [3, n0, _NM,
    0,
    [_NI, _N, _WG, _CT, _T, _LMT],
    [0, 0, 0, 4, 0, 4]
];
var NotebookSessionSummary$ = [3, n0, _NSS,
    0,
    [_SI, _CT],
    [0, 4]
];
var PreparedStatement$ = [3, n0, _PSr,
    0,
    [_SNt, _QSu, _WGNo, _D, _LMT],
    [0, 0, 0, 0, 4]
];
var PreparedStatementSummary$ = [3, n0, _PSS,
    0,
    [_SNt, _LMT],
    [0, 4]
];
var PutCapacityAssignmentConfigurationInput$ = [3, n0, _PCACI,
    0,
    [_CRN, _CAap],
    [0, () => CapacityAssignmentsList], 2
];
var PutCapacityAssignmentConfigurationOutput$ = [3, n0, _PCACO,
    0,
    [],
    []
];
var QueryExecution$ = [3, n0, _QEu,
    0,
    [_QEIu, _Q, _ST, _MQRC, _RC, _RRC, _QEC, _St, _Sta, _WG, _EV, _EP, _STu, _QRSAGC],
    [0, 0, 0, () => ManagedQueryResultsConfiguration$, () => ResultConfiguration$, () => ResultReuseConfiguration$, () => QueryExecutionContext$, () => QueryExecutionStatus$, () => QueryExecutionStatistics$, 0, () => EngineVersion$, 64 | 0, 0, () => QueryResultsS3AccessGrantsConfiguration$]
];
var QueryExecutionContext$ = [3, n0, _QEC,
    0,
    [_Da, _Cat],
    [0, 0]
];
var QueryExecutionStatistics$ = [3, n0, _QES,
    0,
    [_EETIM, _DSIB, _DML, _TETIM, _QQTIM, _SPPTIM, _QPTIM, _SPTIM, _RRI, _DCp],
    [1, 1, 0, 1, 1, 1, 1, 1, () => ResultReuseInformation$, 1]
];
var QueryExecutionStatus$ = [3, n0, _QESu,
    0,
    [_S, _SCR, _SDT, _CDT, _AE],
    [0, 0, 4, 4, () => AthenaError$]
];
var QueryResultsS3AccessGrantsConfiguration$ = [3, n0, _QRSAGC,
    0,
    [_ESAG, _ATu, _CULP],
    [2, 0, 2], 2
];
var QueryRuntimeStatistics$ = [3, n0, _QRS,
    0,
    [_Ti, _Ro, _OS],
    [() => QueryRuntimeStatisticsTimeline$, () => QueryRuntimeStatisticsRows$, () => QueryStage$]
];
var QueryRuntimeStatisticsRows$ = [3, n0, _QRSR,
    0,
    [_IR, _IB, _OB, _OR],
    [1, 1, 1, 1]
];
var QueryRuntimeStatisticsTimeline$ = [3, n0, _QRST,
    0,
    [_QQTIM, _SPPTIM, _QPTIM, _EETIM, _SPTIM, _TETIM],
    [1, 1, 1, 1, 1, 1]
];
var QueryStage$ = [3, n0, _QSue,
    0,
    [_SIt, _S, _OB, _OR, _IB, _IR, _ETxe, _QSP, _SS],
    [1, 0, 1, 1, 1, 1, 1, () => QueryStagePlanNode$, () => QueryStages]
];
var QueryStagePlanNode$ = [3, n0, _QSPN,
    0,
    [_N, _I, _Ch, _RSe],
    [0, 0, () => QueryStagePlanNodes, 64 | 0]
];
var ResourceNotFoundException$ = [-3, n0, _RNFE,
    { [_e]: _c },
    [_M, _RN],
    [0, 0]
];
schema.TypeRegistry.for(n0).registerError(ResourceNotFoundException$, ResourceNotFoundException);
var ResultConfiguration$ = [3, n0, _RC,
    0,
    [_OL, _ECn, _EBO, _AC],
    [0, () => EncryptionConfiguration$, 0, () => AclConfiguration$]
];
var ResultConfigurationUpdates$ = [3, n0, _RCU,
    0,
    [_OL, _ROL, _ECn, _REC, _EBO, _REBO, _AC, _RAC],
    [0, 2, () => EncryptionConfiguration$, 2, 0, 2, () => AclConfiguration$, 2]
];
var ResultReuseByAgeConfiguration$ = [3, n0, _RRBAC,
    0,
    [_E, _MAIM],
    [2, 1], 1
];
var ResultReuseConfiguration$ = [3, n0, _RRC,
    0,
    [_RRBAC],
    [() => ResultReuseByAgeConfiguration$]
];
var ResultReuseInformation$ = [3, n0, _RRI,
    0,
    [_RPR],
    [2], 1
];
var ResultSet$ = [3, n0, _RS,
    0,
    [_Ro, _RSM],
    [() => RowList, () => ResultSetMetadata$]
];
var ResultSetMetadata$ = [3, n0, _RSM,
    0,
    [_CI],
    [() => ColumnInfoList]
];
var Row$ = [3, n0, _Row,
    0,
    [_Data],
    [() => datumList]
];
var S3LoggingConfiguration$ = [3, n0, _SLC,
    0,
    [_E, _KK, _LL],
    [2, 0, 0], 1
];
var SessionAlreadyExistsException$ = [-3, n0, _SAEE,
    { [_e]: _c },
    [_M],
    [0]
];
schema.TypeRegistry.for(n0).registerError(SessionAlreadyExistsException$, SessionAlreadyExistsException);
var SessionConfiguration$ = [3, n0, _SC,
    0,
    [_ER, _WD, _ITS, _SITIM, _ECn],
    [0, 0, 1, 1, () => EncryptionConfiguration$]
];
var SessionStatistics$ = [3, n0, _SSe,
    0,
    [_DEIM],
    [1]
];
var SessionStatus$ = [3, n0, _SSes,
    0,
    [_SDTt, _LMDT, _EDT, _ISDT, _S, _SCR],
    [4, 4, 4, 4, 0, 0]
];
var SessionSummary$ = [3, n0, _SSess,
    0,
    [_SI, _D, _EV, _NV, _St],
    [0, 0, () => EngineVersion$, 0, () => SessionStatus$]
];
var StartCalculationExecutionRequest$ = [3, n0, _SCER,
    0,
    [_SI, _D, _CC, _CB, _CRT],
    [0, 0, () => CalculationConfiguration$, 0, 0], 1
];
var StartCalculationExecutionResponse$ = [3, n0, _SCERt,
    0,
    [_CEI, _S],
    [0, 0]
];
var StartQueryExecutionInput$ = [3, n0, _SQEI,
    0,
    [_QS, _CRT, _QEC, _RC, _WG, _EP, _RRC, _ECng],
    [0, [0, 4], () => QueryExecutionContext$, () => ResultConfiguration$, 0, 64 | 0, () => ResultReuseConfiguration$, () => EngineConfiguration$], 1
];
var StartQueryExecutionOutput$ = [3, n0, _SQEO,
    0,
    [_QEIu],
    [0]
];
var StartSessionRequest$ = [3, n0, _SSR,
    0,
    [_WG, _ECng, _D, _ER, _MC, _NV, _SITIM, _CRT, _Ta, _CWGT],
    [0, () => EngineConfiguration$, 0, 0, () => MonitoringConfiguration$, 0, 1, 0, () => TagList, 2], 2
];
var StartSessionResponse$ = [3, n0, _SSRt,
    0,
    [_SI, _S],
    [0, 0]
];
var StopCalculationExecutionRequest$ = [3, n0, _SCERto,
    0,
    [_CEI],
    [0], 1
];
var StopCalculationExecutionResponse$ = [3, n0, _SCERtop,
    0,
    [_S],
    [0]
];
var StopQueryExecutionInput$ = [3, n0, _SQEIt,
    0,
    [_QEIu],
    [[0, 4]]
];
var StopQueryExecutionOutput$ = [3, n0, _SQEOt,
    0,
    [],
    []
];
var TableMetadata$ = [3, n0, _TM,
    0,
    [_N, _CTr, _LAT, _TT, _Col, _PK, _Pa],
    [0, 4, 4, 0, () => ColumnList, () => ColumnList, 128 | 0], 1
];
var Tag$ = [3, n0, _Tag,
    0,
    [_K, _V],
    [0, 0]
];
var TagResourceInput$ = [3, n0, _TRI,
    0,
    [_RARN, _Ta],
    [0, () => TagList], 2
];
var TagResourceOutput$ = [3, n0, _TRO,
    0,
    [],
    []
];
var TerminateSessionRequest$ = [3, n0, _TSR,
    0,
    [_SI],
    [0], 1
];
var TerminateSessionResponse$ = [3, n0, _TSRe,
    0,
    [_S],
    [0]
];
var TooManyRequestsException$ = [-3, n0, _TMRE,
    { [_e]: _c },
    [_M, _Rea],
    [0, 0]
];
schema.TypeRegistry.for(n0).registerError(TooManyRequestsException$, TooManyRequestsException);
var UnprocessedNamedQueryId$ = [3, n0, _UNQIn,
    0,
    [_NQIa, _ECr, _EM],
    [0, 0, 0]
];
var UnprocessedPreparedStatementName$ = [3, n0, _UPSNn,
    0,
    [_SNt, _ECr, _EM],
    [0, 0, 0]
];
var UnprocessedQueryExecutionId$ = [3, n0, _UQEIn,
    0,
    [_QEIu, _ECr, _EM],
    [0, 0, 0]
];
var UntagResourceInput$ = [3, n0, _URI,
    0,
    [_RARN, _TK],
    [0, 64 | 0], 2
];
var UntagResourceOutput$ = [3, n0, _URO,
    0,
    [],
    []
];
var UpdateCapacityReservationInput$ = [3, n0, _UCRI,
    0,
    [_TD, _N],
    [1, 0], 2
];
var UpdateCapacityReservationOutput$ = [3, n0, _UCRO,
    0,
    [],
    []
];
var UpdateDataCatalogInput$ = [3, n0, _UDCI,
    0,
    [_N, _T, _D, _Pa],
    [0, 0, 0, 128 | 0], 2
];
var UpdateDataCatalogOutput$ = [3, n0, _UDCO,
    0,
    [],
    []
];
var UpdateNamedQueryInput$ = [3, n0, _UNQIp,
    0,
    [_NQIa, _N, _QS, _D],
    [0, 0, 0, 0], 3
];
var UpdateNamedQueryOutput$ = [3, n0, _UNQO,
    0,
    [],
    []
];
var UpdateNotebookInput$ = [3, n0, _UNI,
    0,
    [_NI, _Pay, _T, _SI, _CRT],
    [0, 0, 0, 0, 0], 3
];
var UpdateNotebookMetadataInput$ = [3, n0, _UNMI,
    0,
    [_NI, _N, _CRT],
    [0, 0, 0], 2
];
var UpdateNotebookMetadataOutput$ = [3, n0, _UNMO,
    0,
    [],
    []
];
var UpdateNotebookOutput$ = [3, n0, _UNO,
    0,
    [],
    []
];
var UpdatePreparedStatementInput$ = [3, n0, _UPSI,
    0,
    [_SNt, _WG, _QSu, _D],
    [0, 0, 0, 0], 3
];
var UpdatePreparedStatementOutput$ = [3, n0, _UPSO,
    0,
    [],
    []
];
var UpdateWorkGroupInput$ = [3, n0, _UWGI,
    0,
    [_WG, _D, _CU, _S],
    [0, 0, () => WorkGroupConfigurationUpdates$, 0], 1
];
var UpdateWorkGroupOutput$ = [3, n0, _UWGO,
    0,
    [],
    []
];
var WorkGroup$ = [3, n0, _WG,
    0,
    [_N, _S, _Con, _D, _CT, _ICAA],
    [0, 0, () => WorkGroupConfiguration$, 0, 4, 0], 1
];
var WorkGroupConfiguration$ = [3, n0, _WGC,
    0,
    [_RC, _MQRC, _EWGC, _PCWME, _BSCPQ, _RPE, _EV, _ACdd, _ER, _MC, _ECng, _CCEC, _EMEC, _ICC, _QRSAGC],
    [() => ResultConfiguration$, () => ManagedQueryResultsConfiguration$, 2, 2, 1, 2, () => EngineVersion$, 0, 0, () => MonitoringConfiguration$, () => EngineConfiguration$, () => CustomerContentEncryptionConfiguration$, 2, () => IdentityCenterConfiguration$, () => QueryResultsS3AccessGrantsConfiguration$]
];
var WorkGroupConfigurationUpdates$ = [3, n0, _WGCU,
    0,
    [_EWGC, _RCU, _MQRCU, _PCWME, _BSCPQ, _RBSCPQ, _RPE, _EV, _RCCEC, _ACdd, _ER, _CCEC, _EMEC, _QRSAGC, _MC, _ECng],
    [2, () => ResultConfigurationUpdates$, () => ManagedQueryResultsConfigurationUpdates$, 2, 1, 2, 2, () => EngineVersion$, 2, 0, 0, () => CustomerContentEncryptionConfiguration$, 2, () => QueryResultsS3AccessGrantsConfiguration$, () => MonitoringConfiguration$, () => EngineConfiguration$]
];
var WorkGroupSummary$ = [3, n0, _WGS,
    0,
    [_N, _S, _D, _CT, _EV, _ICAA],
    [0, 0, 0, 4, () => EngineVersion$, 0]
];
var AthenaServiceException$ = [-3, _sm, "AthenaServiceException", 0, [], []];
schema.TypeRegistry.for(_sm).registerError(AthenaServiceException$, AthenaServiceException);
var ApplicationDPUSizesList = [1, n0, _ADPUSL,
    0, () => ApplicationDPUSizes$
];
var CalculationsList = [1, n0, _CL,
    0, () => CalculationSummary$
];
var CapacityAssignmentsList = [1, n0, _CAL,
    0, () => CapacityAssignment$
];
var CapacityReservationsList = [1, n0, _CRL,
    0, () => CapacityReservation$
];
var ClassificationList = [1, n0, _CLl,
    0, () => Classification$
];
var ColumnInfoList = [1, n0, _CIL,
    0, () => ColumnInfo$
];
var ColumnList = [1, n0, _CLo,
    0, () => Column$
];
var DatabaseList = [1, n0, _DL,
    0, () => Database$
];
var DataCatalogSummaryList = [1, n0, _DCSL,
    0, () => DataCatalogSummary$
];
var datumList = [1, n0, _dL,
    0, () => Datum$
];
var EngineVersionsList = [1, n0, _EVL,
    0, () => EngineVersion$
];
var ExecutorsSummaryList = [1, n0, _ESL,
    0, () => ExecutorsSummary$
];
var NamedQueryList = [1, n0, _NQL,
    0, () => NamedQuery$
];
var NotebookMetadataArray = [1, n0, _NMA,
    0, () => NotebookMetadata$
];
var NotebookSessionsList = [1, n0, _NSL,
    0, () => NotebookSessionSummary$
];
var PreparedStatementDetailsList = [1, n0, _PSDL,
    0, () => PreparedStatement$
];
var PreparedStatementsList = [1, n0, _PSL,
    0, () => PreparedStatementSummary$
];
var QueryExecutionList = [1, n0, _QEL,
    0, () => QueryExecution$
];
var QueryStagePlanNodes = [1, n0, _QSPNu,
    0, () => QueryStagePlanNode$
];
var QueryStages = [1, n0, _QSuer,
    0, () => QueryStage$
];
var RowList = [1, n0, _RL,
    0, () => Row$
];
var SessionsList = [1, n0, _SL,
    0, () => SessionSummary$
];
var TableMetadataList = [1, n0, _TML,
    0, () => TableMetadata$
];
var TagList = [1, n0, _TL,
    0, () => Tag$
];
var UnprocessedNamedQueryIdList = [1, n0, _UNQIL,
    0, () => UnprocessedNamedQueryId$
];
var UnprocessedPreparedStatementNameList = [1, n0, _UPSNL,
    0, () => UnprocessedPreparedStatementName$
];
var UnprocessedQueryExecutionIdList = [1, n0, _UQEIL,
    0, () => UnprocessedQueryExecutionId$
];
var WorkGroupsList = [1, n0, _WGL,
    0, () => WorkGroupSummary$
];
var BatchGetNamedQuery$ = [9, n0, _BGNQ,
    0, () => BatchGetNamedQueryInput$, () => BatchGetNamedQueryOutput$
];
var BatchGetPreparedStatement$ = [9, n0, _BGPS,
    0, () => BatchGetPreparedStatementInput$, () => BatchGetPreparedStatementOutput$
];
var BatchGetQueryExecution$ = [9, n0, _BGQE,
    0, () => BatchGetQueryExecutionInput$, () => BatchGetQueryExecutionOutput$
];
var CancelCapacityReservation$ = [9, n0, _CCR,
    2, () => CancelCapacityReservationInput$, () => CancelCapacityReservationOutput$
];
var CreateCapacityReservation$ = [9, n0, _CCRr,
    2, () => CreateCapacityReservationInput$, () => CreateCapacityReservationOutput$
];
var CreateDataCatalog$ = [9, n0, _CDC,
    0, () => CreateDataCatalogInput$, () => CreateDataCatalogOutput$
];
var CreateNamedQuery$ = [9, n0, _CNQ,
    2, () => CreateNamedQueryInput$, () => CreateNamedQueryOutput$
];
var CreateNotebook$ = [9, n0, _CNr,
    0, () => CreateNotebookInput$, () => CreateNotebookOutput$
];
var CreatePreparedStatement$ = [9, n0, _CPS,
    0, () => CreatePreparedStatementInput$, () => CreatePreparedStatementOutput$
];
var CreatePresignedNotebookUrl$ = [9, n0, _CPNU,
    0, () => CreatePresignedNotebookUrlRequest$, () => CreatePresignedNotebookUrlResponse$
];
var CreateWorkGroup$ = [9, n0, _CWG,
    0, () => CreateWorkGroupInput$, () => CreateWorkGroupOutput$
];
var DeleteCapacityReservation$ = [9, n0, _DCR,
    2, () => DeleteCapacityReservationInput$, () => DeleteCapacityReservationOutput$
];
var DeleteDataCatalog$ = [9, n0, _DDC,
    0, () => DeleteDataCatalogInput$, () => DeleteDataCatalogOutput$
];
var DeleteNamedQuery$ = [9, n0, _DNQ,
    2, () => DeleteNamedQueryInput$, () => DeleteNamedQueryOutput$
];
var DeleteNotebook$ = [9, n0, _DNe,
    0, () => DeleteNotebookInput$, () => DeleteNotebookOutput$
];
var DeletePreparedStatement$ = [9, n0, _DPS,
    0, () => DeletePreparedStatementInput$, () => DeletePreparedStatementOutput$
];
var DeleteWorkGroup$ = [9, n0, _DWG,
    2, () => DeleteWorkGroupInput$, () => DeleteWorkGroupOutput$
];
var ExportNotebook$ = [9, n0, _EN,
    0, () => ExportNotebookInput$, () => ExportNotebookOutput$
];
var GetCalculationExecution$ = [9, n0, _GCE,
    0, () => GetCalculationExecutionRequest$, () => GetCalculationExecutionResponse$
];
var GetCalculationExecutionCode$ = [9, n0, _GCEC,
    0, () => GetCalculationExecutionCodeRequest$, () => GetCalculationExecutionCodeResponse$
];
var GetCalculationExecutionStatus$ = [9, n0, _GCES,
    0, () => GetCalculationExecutionStatusRequest$, () => GetCalculationExecutionStatusResponse$
];
var GetCapacityAssignmentConfiguration$ = [9, n0, _GCAC,
    0, () => GetCapacityAssignmentConfigurationInput$, () => GetCapacityAssignmentConfigurationOutput$
];
var GetCapacityReservation$ = [9, n0, _GCR,
    0, () => GetCapacityReservationInput$, () => GetCapacityReservationOutput$
];
var GetDatabase$ = [9, n0, _GD,
    0, () => GetDatabaseInput$, () => GetDatabaseOutput$
];
var GetDataCatalog$ = [9, n0, _GDC,
    0, () => GetDataCatalogInput$, () => GetDataCatalogOutput$
];
var GetNamedQuery$ = [9, n0, _GNQ,
    0, () => GetNamedQueryInput$, () => GetNamedQueryOutput$
];
var GetNotebookMetadata$ = [9, n0, _GNM,
    0, () => GetNotebookMetadataInput$, () => GetNotebookMetadataOutput$
];
var GetPreparedStatement$ = [9, n0, _GPS,
    0, () => GetPreparedStatementInput$, () => GetPreparedStatementOutput$
];
var GetQueryExecution$ = [9, n0, _GQE,
    0, () => GetQueryExecutionInput$, () => GetQueryExecutionOutput$
];
var GetQueryResults$ = [9, n0, _GQR,
    0, () => GetQueryResultsInput$, () => GetQueryResultsOutput$
];
var GetQueryRuntimeStatistics$ = [9, n0, _GQRS,
    0, () => GetQueryRuntimeStatisticsInput$, () => GetQueryRuntimeStatisticsOutput$
];
var GetResourceDashboard$ = [9, n0, _GRD,
    0, () => GetResourceDashboardRequest$, () => GetResourceDashboardResponse$
];
var GetSession$ = [9, n0, _GS,
    0, () => GetSessionRequest$, () => GetSessionResponse$
];
var GetSessionEndpoint$ = [9, n0, _GSE,
    0, () => GetSessionEndpointRequest$, () => GetSessionEndpointResponse$
];
var GetSessionStatus$ = [9, n0, _GSS,
    0, () => GetSessionStatusRequest$, () => GetSessionStatusResponse$
];
var GetTableMetadata$ = [9, n0, _GTM,
    0, () => GetTableMetadataInput$, () => GetTableMetadataOutput$
];
var GetWorkGroup$ = [9, n0, _GWG,
    0, () => GetWorkGroupInput$, () => GetWorkGroupOutput$
];
var ImportNotebook$ = [9, n0, _IN,
    0, () => ImportNotebookInput$, () => ImportNotebookOutput$
];
var ListApplicationDPUSizes$ = [9, n0, _LADPUS,
    0, () => ListApplicationDPUSizesInput$, () => ListApplicationDPUSizesOutput$
];
var ListCalculationExecutions$ = [9, n0, _LCE,
    0, () => ListCalculationExecutionsRequest$, () => ListCalculationExecutionsResponse$
];
var ListCapacityReservations$ = [9, n0, _LCR,
    0, () => ListCapacityReservationsInput$, () => ListCapacityReservationsOutput$
];
var ListDatabases$ = [9, n0, _LD,
    0, () => ListDatabasesInput$, () => ListDatabasesOutput$
];
var ListDataCatalogs$ = [9, n0, _LDC,
    0, () => ListDataCatalogsInput$, () => ListDataCatalogsOutput$
];
var ListEngineVersions$ = [9, n0, _LEV,
    0, () => ListEngineVersionsInput$, () => ListEngineVersionsOutput$
];
var ListExecutors$ = [9, n0, _LE,
    0, () => ListExecutorsRequest$, () => ListExecutorsResponse$
];
var ListNamedQueries$ = [9, n0, _LNQ,
    0, () => ListNamedQueriesInput$, () => ListNamedQueriesOutput$
];
var ListNotebookMetadata$ = [9, n0, _LNM,
    0, () => ListNotebookMetadataInput$, () => ListNotebookMetadataOutput$
];
var ListNotebookSessions$ = [9, n0, _LNS,
    0, () => ListNotebookSessionsRequest$, () => ListNotebookSessionsResponse$
];
var ListPreparedStatements$ = [9, n0, _LPS,
    0, () => ListPreparedStatementsInput$, () => ListPreparedStatementsOutput$
];
var ListQueryExecutions$ = [9, n0, _LQE,
    0, () => ListQueryExecutionsInput$, () => ListQueryExecutionsOutput$
];
var ListSessions$ = [9, n0, _LS,
    0, () => ListSessionsRequest$, () => ListSessionsResponse$
];
var ListTableMetadata$ = [9, n0, _LTMi,
    0, () => ListTableMetadataInput$, () => ListTableMetadataOutput$
];
var ListTagsForResource$ = [9, n0, _LTFR,
    0, () => ListTagsForResourceInput$, () => ListTagsForResourceOutput$
];
var ListWorkGroups$ = [9, n0, _LWG,
    0, () => ListWorkGroupsInput$, () => ListWorkGroupsOutput$
];
var PutCapacityAssignmentConfiguration$ = [9, n0, _PCAC,
    2, () => PutCapacityAssignmentConfigurationInput$, () => PutCapacityAssignmentConfigurationOutput$
];
var StartCalculationExecution$ = [9, n0, _SCE,
    0, () => StartCalculationExecutionRequest$, () => StartCalculationExecutionResponse$
];
var StartQueryExecution$ = [9, n0, _SQE,
    2, () => StartQueryExecutionInput$, () => StartQueryExecutionOutput$
];
var StartSession$ = [9, n0, _SSt,
    0, () => StartSessionRequest$, () => StartSessionResponse$
];
var StopCalculationExecution$ = [9, n0, _SCEt,
    0, () => StopCalculationExecutionRequest$, () => StopCalculationExecutionResponse$
];
var StopQueryExecution$ = [9, n0, _SQEt,
    2, () => StopQueryExecutionInput$, () => StopQueryExecutionOutput$
];
var TagResource$ = [9, n0, _TR,
    0, () => TagResourceInput$, () => TagResourceOutput$
];
var TerminateSession$ = [9, n0, _TS,
    0, () => TerminateSessionRequest$, () => TerminateSessionResponse$
];
var UntagResource$ = [9, n0, _UR,
    0, () => UntagResourceInput$, () => UntagResourceOutput$
];
var UpdateCapacityReservation$ = [9, n0, _UCR,
    0, () => UpdateCapacityReservationInput$, () => UpdateCapacityReservationOutput$
];
var UpdateDataCatalog$ = [9, n0, _UDC,
    0, () => UpdateDataCatalogInput$, () => UpdateDataCatalogOutput$
];
var UpdateNamedQuery$ = [9, n0, _UNQ,
    2, () => UpdateNamedQueryInput$, () => UpdateNamedQueryOutput$
];
var UpdateNotebook$ = [9, n0, _UN,
    0, () => UpdateNotebookInput$, () => UpdateNotebookOutput$
];
var UpdateNotebookMetadata$ = [9, n0, _UNM,
    0, () => UpdateNotebookMetadataInput$, () => UpdateNotebookMetadataOutput$
];
var UpdatePreparedStatement$ = [9, n0, _UPS,
    0, () => UpdatePreparedStatementInput$, () => UpdatePreparedStatementOutput$
];
var UpdateWorkGroup$ = [9, n0, _UWG,
    0, () => UpdateWorkGroupInput$, () => UpdateWorkGroupOutput$
];

class BatchGetNamedQueryCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonAthena", "BatchGetNamedQuery", {})
    .n("AthenaClient", "BatchGetNamedQueryCommand")
    .sc(BatchGetNamedQuery$)
    .build() {
}

class BatchGetPreparedStatementCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonAthena", "BatchGetPreparedStatement", {})
    .n("AthenaClient", "BatchGetPreparedStatementCommand")
    .sc(BatchGetPreparedStatement$)
    .build() {
}

class BatchGetQueryExecutionCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonAthena", "BatchGetQueryExecution", {})
    .n("AthenaClient", "BatchGetQueryExecutionCommand")
    .sc(BatchGetQueryExecution$)
    .build() {
}

class CancelCapacityReservationCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonAthena", "CancelCapacityReservation", {})
    .n("AthenaClient", "CancelCapacityReservationCommand")
    .sc(CancelCapacityReservation$)
    .build() {
}

class CreateCapacityReservationCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonAthena", "CreateCapacityReservation", {})
    .n("AthenaClient", "CreateCapacityReservationCommand")
    .sc(CreateCapacityReservation$)
    .build() {
}

class CreateDataCatalogCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonAthena", "CreateDataCatalog", {})
    .n("AthenaClient", "CreateDataCatalogCommand")
    .sc(CreateDataCatalog$)
    .build() {
}

class CreateNamedQueryCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonAthena", "CreateNamedQuery", {})
    .n("AthenaClient", "CreateNamedQueryCommand")
    .sc(CreateNamedQuery$)
    .build() {
}

class CreateNotebookCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonAthena", "CreateNotebook", {})
    .n("AthenaClient", "CreateNotebookCommand")
    .sc(CreateNotebook$)
    .build() {
}

class CreatePreparedStatementCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonAthena", "CreatePreparedStatement", {})
    .n("AthenaClient", "CreatePreparedStatementCommand")
    .sc(CreatePreparedStatement$)
    .build() {
}

class CreatePresignedNotebookUrlCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonAthena", "CreatePresignedNotebookUrl", {})
    .n("AthenaClient", "CreatePresignedNotebookUrlCommand")
    .sc(CreatePresignedNotebookUrl$)
    .build() {
}

class CreateWorkGroupCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonAthena", "CreateWorkGroup", {})
    .n("AthenaClient", "CreateWorkGroupCommand")
    .sc(CreateWorkGroup$)
    .build() {
}

class DeleteCapacityReservationCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonAthena", "DeleteCapacityReservation", {})
    .n("AthenaClient", "DeleteCapacityReservationCommand")
    .sc(DeleteCapacityReservation$)
    .build() {
}

class DeleteDataCatalogCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonAthena", "DeleteDataCatalog", {})
    .n("AthenaClient", "DeleteDataCatalogCommand")
    .sc(DeleteDataCatalog$)
    .build() {
}

class DeleteNamedQueryCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonAthena", "DeleteNamedQuery", {})
    .n("AthenaClient", "DeleteNamedQueryCommand")
    .sc(DeleteNamedQuery$)
    .build() {
}

class DeleteNotebookCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonAthena", "DeleteNotebook", {})
    .n("AthenaClient", "DeleteNotebookCommand")
    .sc(DeleteNotebook$)
    .build() {
}

class DeletePreparedStatementCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonAthena", "DeletePreparedStatement", {})
    .n("AthenaClient", "DeletePreparedStatementCommand")
    .sc(DeletePreparedStatement$)
    .build() {
}

class DeleteWorkGroupCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonAthena", "DeleteWorkGroup", {})
    .n("AthenaClient", "DeleteWorkGroupCommand")
    .sc(DeleteWorkGroup$)
    .build() {
}

class ExportNotebookCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonAthena", "ExportNotebook", {})
    .n("AthenaClient", "ExportNotebookCommand")
    .sc(ExportNotebook$)
    .build() {
}

class GetCalculationExecutionCodeCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonAthena", "GetCalculationExecutionCode", {})
    .n("AthenaClient", "GetCalculationExecutionCodeCommand")
    .sc(GetCalculationExecutionCode$)
    .build() {
}

class GetCalculationExecutionCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonAthena", "GetCalculationExecution", {})
    .n("AthenaClient", "GetCalculationExecutionCommand")
    .sc(GetCalculationExecution$)
    .build() {
}

class GetCalculationExecutionStatusCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonAthena", "GetCalculationExecutionStatus", {})
    .n("AthenaClient", "GetCalculationExecutionStatusCommand")
    .sc(GetCalculationExecutionStatus$)
    .build() {
}

class GetCapacityAssignmentConfigurationCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonAthena", "GetCapacityAssignmentConfiguration", {})
    .n("AthenaClient", "GetCapacityAssignmentConfigurationCommand")
    .sc(GetCapacityAssignmentConfiguration$)
    .build() {
}

class GetCapacityReservationCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonAthena", "GetCapacityReservation", {})
    .n("AthenaClient", "GetCapacityReservationCommand")
    .sc(GetCapacityReservation$)
    .build() {
}

class GetDatabaseCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonAthena", "GetDatabase", {})
    .n("AthenaClient", "GetDatabaseCommand")
    .sc(GetDatabase$)
    .build() {
}

class GetDataCatalogCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonAthena", "GetDataCatalog", {})
    .n("AthenaClient", "GetDataCatalogCommand")
    .sc(GetDataCatalog$)
    .build() {
}

class GetNamedQueryCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonAthena", "GetNamedQuery", {})
    .n("AthenaClient", "GetNamedQueryCommand")
    .sc(GetNamedQuery$)
    .build() {
}

class GetNotebookMetadataCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonAthena", "GetNotebookMetadata", {})
    .n("AthenaClient", "GetNotebookMetadataCommand")
    .sc(GetNotebookMetadata$)
    .build() {
}

class GetPreparedStatementCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonAthena", "GetPreparedStatement", {})
    .n("AthenaClient", "GetPreparedStatementCommand")
    .sc(GetPreparedStatement$)
    .build() {
}

class GetQueryExecutionCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonAthena", "GetQueryExecution", {})
    .n("AthenaClient", "GetQueryExecutionCommand")
    .sc(GetQueryExecution$)
    .build() {
}

class GetQueryResultsCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonAthena", "GetQueryResults", {})
    .n("AthenaClient", "GetQueryResultsCommand")
    .sc(GetQueryResults$)
    .build() {
}

class GetQueryRuntimeStatisticsCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonAthena", "GetQueryRuntimeStatistics", {})
    .n("AthenaClient", "GetQueryRuntimeStatisticsCommand")
    .sc(GetQueryRuntimeStatistics$)
    .build() {
}

class GetResourceDashboardCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonAthena", "GetResourceDashboard", {})
    .n("AthenaClient", "GetResourceDashboardCommand")
    .sc(GetResourceDashboard$)
    .build() {
}

class GetSessionCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonAthena", "GetSession", {})
    .n("AthenaClient", "GetSessionCommand")
    .sc(GetSession$)
    .build() {
}

class GetSessionEndpointCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonAthena", "GetSessionEndpoint", {})
    .n("AthenaClient", "GetSessionEndpointCommand")
    .sc(GetSessionEndpoint$)
    .build() {
}

class GetSessionStatusCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonAthena", "GetSessionStatus", {})
    .n("AthenaClient", "GetSessionStatusCommand")
    .sc(GetSessionStatus$)
    .build() {
}

class GetTableMetadataCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonAthena", "GetTableMetadata", {})
    .n("AthenaClient", "GetTableMetadataCommand")
    .sc(GetTableMetadata$)
    .build() {
}

class GetWorkGroupCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonAthena", "GetWorkGroup", {})
    .n("AthenaClient", "GetWorkGroupCommand")
    .sc(GetWorkGroup$)
    .build() {
}

class ImportNotebookCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonAthena", "ImportNotebook", {})
    .n("AthenaClient", "ImportNotebookCommand")
    .sc(ImportNotebook$)
    .build() {
}

class ListApplicationDPUSizesCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonAthena", "ListApplicationDPUSizes", {})
    .n("AthenaClient", "ListApplicationDPUSizesCommand")
    .sc(ListApplicationDPUSizes$)
    .build() {
}

class ListCalculationExecutionsCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonAthena", "ListCalculationExecutions", {})
    .n("AthenaClient", "ListCalculationExecutionsCommand")
    .sc(ListCalculationExecutions$)
    .build() {
}

class ListCapacityReservationsCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonAthena", "ListCapacityReservations", {})
    .n("AthenaClient", "ListCapacityReservationsCommand")
    .sc(ListCapacityReservations$)
    .build() {
}

class ListDatabasesCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonAthena", "ListDatabases", {})
    .n("AthenaClient", "ListDatabasesCommand")
    .sc(ListDatabases$)
    .build() {
}

class ListDataCatalogsCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonAthena", "ListDataCatalogs", {})
    .n("AthenaClient", "ListDataCatalogsCommand")
    .sc(ListDataCatalogs$)
    .build() {
}

class ListEngineVersionsCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonAthena", "ListEngineVersions", {})
    .n("AthenaClient", "ListEngineVersionsCommand")
    .sc(ListEngineVersions$)
    .build() {
}

class ListExecutorsCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonAthena", "ListExecutors", {})
    .n("AthenaClient", "ListExecutorsCommand")
    .sc(ListExecutors$)
    .build() {
}

class ListNamedQueriesCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonAthena", "ListNamedQueries", {})
    .n("AthenaClient", "ListNamedQueriesCommand")
    .sc(ListNamedQueries$)
    .build() {
}

class ListNotebookMetadataCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonAthena", "ListNotebookMetadata", {})
    .n("AthenaClient", "ListNotebookMetadataCommand")
    .sc(ListNotebookMetadata$)
    .build() {
}

class ListNotebookSessionsCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonAthena", "ListNotebookSessions", {})
    .n("AthenaClient", "ListNotebookSessionsCommand")
    .sc(ListNotebookSessions$)
    .build() {
}

class ListPreparedStatementsCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonAthena", "ListPreparedStatements", {})
    .n("AthenaClient", "ListPreparedStatementsCommand")
    .sc(ListPreparedStatements$)
    .build() {
}

class ListQueryExecutionsCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonAthena", "ListQueryExecutions", {})
    .n("AthenaClient", "ListQueryExecutionsCommand")
    .sc(ListQueryExecutions$)
    .build() {
}

class ListSessionsCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonAthena", "ListSessions", {})
    .n("AthenaClient", "ListSessionsCommand")
    .sc(ListSessions$)
    .build() {
}

class ListTableMetadataCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonAthena", "ListTableMetadata", {})
    .n("AthenaClient", "ListTableMetadataCommand")
    .sc(ListTableMetadata$)
    .build() {
}

class ListTagsForResourceCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonAthena", "ListTagsForResource", {})
    .n("AthenaClient", "ListTagsForResourceCommand")
    .sc(ListTagsForResource$)
    .build() {
}

class ListWorkGroupsCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonAthena", "ListWorkGroups", {})
    .n("AthenaClient", "ListWorkGroupsCommand")
    .sc(ListWorkGroups$)
    .build() {
}

class PutCapacityAssignmentConfigurationCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonAthena", "PutCapacityAssignmentConfiguration", {})
    .n("AthenaClient", "PutCapacityAssignmentConfigurationCommand")
    .sc(PutCapacityAssignmentConfiguration$)
    .build() {
}

class StartCalculationExecutionCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonAthena", "StartCalculationExecution", {})
    .n("AthenaClient", "StartCalculationExecutionCommand")
    .sc(StartCalculationExecution$)
    .build() {
}

class StartQueryExecutionCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonAthena", "StartQueryExecution", {})
    .n("AthenaClient", "StartQueryExecutionCommand")
    .sc(StartQueryExecution$)
    .build() {
}

class StartSessionCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonAthena", "StartSession", {})
    .n("AthenaClient", "StartSessionCommand")
    .sc(StartSession$)
    .build() {
}

class StopCalculationExecutionCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonAthena", "StopCalculationExecution", {})
    .n("AthenaClient", "StopCalculationExecutionCommand")
    .sc(StopCalculationExecution$)
    .build() {
}

class StopQueryExecutionCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonAthena", "StopQueryExecution", {})
    .n("AthenaClient", "StopQueryExecutionCommand")
    .sc(StopQueryExecution$)
    .build() {
}

class TagResourceCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonAthena", "TagResource", {})
    .n("AthenaClient", "TagResourceCommand")
    .sc(TagResource$)
    .build() {
}

class TerminateSessionCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonAthena", "TerminateSession", {})
    .n("AthenaClient", "TerminateSessionCommand")
    .sc(TerminateSession$)
    .build() {
}

class UntagResourceCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonAthena", "UntagResource", {})
    .n("AthenaClient", "UntagResourceCommand")
    .sc(UntagResource$)
    .build() {
}

class UpdateCapacityReservationCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonAthena", "UpdateCapacityReservation", {})
    .n("AthenaClient", "UpdateCapacityReservationCommand")
    .sc(UpdateCapacityReservation$)
    .build() {
}

class UpdateDataCatalogCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonAthena", "UpdateDataCatalog", {})
    .n("AthenaClient", "UpdateDataCatalogCommand")
    .sc(UpdateDataCatalog$)
    .build() {
}

class UpdateNamedQueryCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonAthena", "UpdateNamedQuery", {})
    .n("AthenaClient", "UpdateNamedQueryCommand")
    .sc(UpdateNamedQuery$)
    .build() {
}

class UpdateNotebookCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonAthena", "UpdateNotebook", {})
    .n("AthenaClient", "UpdateNotebookCommand")
    .sc(UpdateNotebook$)
    .build() {
}

class UpdateNotebookMetadataCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonAthena", "UpdateNotebookMetadata", {})
    .n("AthenaClient", "UpdateNotebookMetadataCommand")
    .sc(UpdateNotebookMetadata$)
    .build() {
}

class UpdatePreparedStatementCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonAthena", "UpdatePreparedStatement", {})
    .n("AthenaClient", "UpdatePreparedStatementCommand")
    .sc(UpdatePreparedStatement$)
    .build() {
}

class UpdateWorkGroupCommand extends smithyClient.Command
    .classBuilder()
    .ep(commonParams)
    .m(function (Command, cs, config, o) {
    return [middlewareEndpoint.getEndpointPlugin(config, Command.getEndpointParameterInstructions())];
})
    .s("AmazonAthena", "UpdateWorkGroup", {})
    .n("AthenaClient", "UpdateWorkGroupCommand")
    .sc(UpdateWorkGroup$)
    .build() {
}

const commands = {
    BatchGetNamedQueryCommand,
    BatchGetPreparedStatementCommand,
    BatchGetQueryExecutionCommand,
    CancelCapacityReservationCommand,
    CreateCapacityReservationCommand,
    CreateDataCatalogCommand,
    CreateNamedQueryCommand,
    CreateNotebookCommand,
    CreatePreparedStatementCommand,
    CreatePresignedNotebookUrlCommand,
    CreateWorkGroupCommand,
    DeleteCapacityReservationCommand,
    DeleteDataCatalogCommand,
    DeleteNamedQueryCommand,
    DeleteNotebookCommand,
    DeletePreparedStatementCommand,
    DeleteWorkGroupCommand,
    ExportNotebookCommand,
    GetCalculationExecutionCommand,
    GetCalculationExecutionCodeCommand,
    GetCalculationExecutionStatusCommand,
    GetCapacityAssignmentConfigurationCommand,
    GetCapacityReservationCommand,
    GetDatabaseCommand,
    GetDataCatalogCommand,
    GetNamedQueryCommand,
    GetNotebookMetadataCommand,
    GetPreparedStatementCommand,
    GetQueryExecutionCommand,
    GetQueryResultsCommand,
    GetQueryRuntimeStatisticsCommand,
    GetResourceDashboardCommand,
    GetSessionCommand,
    GetSessionEndpointCommand,
    GetSessionStatusCommand,
    GetTableMetadataCommand,
    GetWorkGroupCommand,
    ImportNotebookCommand,
    ListApplicationDPUSizesCommand,
    ListCalculationExecutionsCommand,
    ListCapacityReservationsCommand,
    ListDatabasesCommand,
    ListDataCatalogsCommand,
    ListEngineVersionsCommand,
    ListExecutorsCommand,
    ListNamedQueriesCommand,
    ListNotebookMetadataCommand,
    ListNotebookSessionsCommand,
    ListPreparedStatementsCommand,
    ListQueryExecutionsCommand,
    ListSessionsCommand,
    ListTableMetadataCommand,
    ListTagsForResourceCommand,
    ListWorkGroupsCommand,
    PutCapacityAssignmentConfigurationCommand,
    StartCalculationExecutionCommand,
    StartQueryExecutionCommand,
    StartSessionCommand,
    StopCalculationExecutionCommand,
    StopQueryExecutionCommand,
    TagResourceCommand,
    TerminateSessionCommand,
    UntagResourceCommand,
    UpdateCapacityReservationCommand,
    UpdateDataCatalogCommand,
    UpdateNamedQueryCommand,
    UpdateNotebookCommand,
    UpdateNotebookMetadataCommand,
    UpdatePreparedStatementCommand,
    UpdateWorkGroupCommand,
};
class Athena extends AthenaClient {
}
smithyClient.createAggregatedClient(commands, Athena);

const paginateGetQueryResults = core.createPaginator(AthenaClient, GetQueryResultsCommand, "NextToken", "NextToken", "MaxResults");

const paginateListApplicationDPUSizes = core.createPaginator(AthenaClient, ListApplicationDPUSizesCommand, "NextToken", "NextToken", "MaxResults");

const paginateListCalculationExecutions = core.createPaginator(AthenaClient, ListCalculationExecutionsCommand, "NextToken", "NextToken", "MaxResults");

const paginateListCapacityReservations = core.createPaginator(AthenaClient, ListCapacityReservationsCommand, "NextToken", "NextToken", "MaxResults");

const paginateListDatabases = core.createPaginator(AthenaClient, ListDatabasesCommand, "NextToken", "NextToken", "MaxResults");

const paginateListDataCatalogs = core.createPaginator(AthenaClient, ListDataCatalogsCommand, "NextToken", "NextToken", "MaxResults");

const paginateListEngineVersions = core.createPaginator(AthenaClient, ListEngineVersionsCommand, "NextToken", "NextToken", "MaxResults");

const paginateListExecutors = core.createPaginator(AthenaClient, ListExecutorsCommand, "NextToken", "NextToken", "MaxResults");

const paginateListNamedQueries = core.createPaginator(AthenaClient, ListNamedQueriesCommand, "NextToken", "NextToken", "MaxResults");

const paginateListPreparedStatements = core.createPaginator(AthenaClient, ListPreparedStatementsCommand, "NextToken", "NextToken", "MaxResults");

const paginateListQueryExecutions = core.createPaginator(AthenaClient, ListQueryExecutionsCommand, "NextToken", "NextToken", "MaxResults");

const paginateListSessions = core.createPaginator(AthenaClient, ListSessionsCommand, "NextToken", "NextToken", "MaxResults");

const paginateListTableMetadata = core.createPaginator(AthenaClient, ListTableMetadataCommand, "NextToken", "NextToken", "MaxResults");

const paginateListTagsForResource = core.createPaginator(AthenaClient, ListTagsForResourceCommand, "NextToken", "NextToken", "MaxResults");

const paginateListWorkGroups = core.createPaginator(AthenaClient, ListWorkGroupsCommand, "NextToken", "NextToken", "MaxResults");

const S3AclOption = {
    BUCKET_OWNER_FULL_CONTROL: "BUCKET_OWNER_FULL_CONTROL",
};
const AuthenticationType = {
    DIRECTORY_IDENTITY: "DIRECTORY_IDENTITY",
};
const EncryptionOption = {
    CSE_KMS: "CSE_KMS",
    SSE_KMS: "SSE_KMS",
    SSE_S3: "SSE_S3",
};
const StatementType = {
    DDL: "DDL",
    DML: "DML",
    UTILITY: "UTILITY",
};
const QueryExecutionState = {
    CANCELLED: "CANCELLED",
    FAILED: "FAILED",
    QUEUED: "QUEUED",
    RUNNING: "RUNNING",
    SUCCEEDED: "SUCCEEDED",
};
const DataCatalogType = {
    FEDERATED: "FEDERATED",
    GLUE: "GLUE",
    HIVE: "HIVE",
    LAMBDA: "LAMBDA",
};
const ConnectionType = {
    BIGQUERY: "BIGQUERY",
    CMDB: "CMDB",
    DATALAKEGEN2: "DATALAKEGEN2",
    DB2: "DB2",
    DB2AS400: "DB2AS400",
    DOCUMENTDB: "DOCUMENTDB",
    DYNAMODB: "DYNAMODB",
    GOOGLECLOUDSTORAGE: "GOOGLECLOUDSTORAGE",
    HBASE: "HBASE",
    MYSQL: "MYSQL",
    OPENSEARCH: "OPENSEARCH",
    ORACLE: "ORACLE",
    POSTGRESQL: "POSTGRESQL",
    REDSHIFT: "REDSHIFT",
    SAPHANA: "SAPHANA",
    SNOWFLAKE: "SNOWFLAKE",
    SQLSERVER: "SQLSERVER",
    SYNAPSE: "SYNAPSE",
    TIMESTREAM: "TIMESTREAM",
    TPCDS: "TPCDS",
};
const DataCatalogStatus = {
    CREATE_COMPLETE: "CREATE_COMPLETE",
    CREATE_FAILED: "CREATE_FAILED",
    CREATE_FAILED_CLEANUP_COMPLETE: "CREATE_FAILED_CLEANUP_COMPLETE",
    CREATE_FAILED_CLEANUP_FAILED: "CREATE_FAILED_CLEANUP_FAILED",
    CREATE_FAILED_CLEANUP_IN_PROGRESS: "CREATE_FAILED_CLEANUP_IN_PROGRESS",
    CREATE_IN_PROGRESS: "CREATE_IN_PROGRESS",
    DELETE_COMPLETE: "DELETE_COMPLETE",
    DELETE_FAILED: "DELETE_FAILED",
    DELETE_IN_PROGRESS: "DELETE_IN_PROGRESS",
};
const ThrottleReason = {
    CONCURRENT_QUERY_LIMIT_EXCEEDED: "CONCURRENT_QUERY_LIMIT_EXCEEDED",
};
const NotebookType = {
    IPYNB: "IPYNB",
};
const CalculationExecutionState = {
    CANCELED: "CANCELED",
    CANCELING: "CANCELING",
    COMPLETED: "COMPLETED",
    CREATED: "CREATED",
    CREATING: "CREATING",
    FAILED: "FAILED",
    QUEUED: "QUEUED",
    RUNNING: "RUNNING",
};
const CapacityAllocationStatus = {
    FAILED: "FAILED",
    PENDING: "PENDING",
    SUCCEEDED: "SUCCEEDED",
};
const CapacityReservationStatus = {
    ACTIVE: "ACTIVE",
    CANCELLED: "CANCELLED",
    CANCELLING: "CANCELLING",
    FAILED: "FAILED",
    PENDING: "PENDING",
    UPDATE_PENDING: "UPDATE_PENDING",
};
const QueryResultType = {
    DATA_MANIFEST: "DATA_MANIFEST",
    DATA_ROWS: "DATA_ROWS",
};
const ColumnNullable = {
    NOT_NULL: "NOT_NULL",
    NULLABLE: "NULLABLE",
    UNKNOWN: "UNKNOWN",
};
const SessionState = {
    BUSY: "BUSY",
    CREATED: "CREATED",
    CREATING: "CREATING",
    DEGRADED: "DEGRADED",
    FAILED: "FAILED",
    IDLE: "IDLE",
    TERMINATED: "TERMINATED",
    TERMINATING: "TERMINATING",
};
const WorkGroupState = {
    DISABLED: "DISABLED",
    ENABLED: "ENABLED",
};
const ExecutorState = {
    CREATED: "CREATED",
    CREATING: "CREATING",
    FAILED: "FAILED",
    REGISTERED: "REGISTERED",
    TERMINATED: "TERMINATED",
    TERMINATING: "TERMINATING",
};
const ExecutorType = {
    COORDINATOR: "COORDINATOR",
    GATEWAY: "GATEWAY",
    WORKER: "WORKER",
};

Object.defineProperty(exports, "$Command", {
    enumerable: true,
    get: function () { return smithyClient.Command; }
});
Object.defineProperty(exports, "__Client", {
    enumerable: true,
    get: function () { return smithyClient.Client; }
});
exports.AclConfiguration$ = AclConfiguration$;
exports.ApplicationDPUSizes$ = ApplicationDPUSizes$;
exports.Athena = Athena;
exports.AthenaClient = AthenaClient;
exports.AthenaError$ = AthenaError$;
exports.AthenaServiceException = AthenaServiceException;
exports.AthenaServiceException$ = AthenaServiceException$;
exports.AuthenticationType = AuthenticationType;
exports.BatchGetNamedQuery$ = BatchGetNamedQuery$;
exports.BatchGetNamedQueryCommand = BatchGetNamedQueryCommand;
exports.BatchGetNamedQueryInput$ = BatchGetNamedQueryInput$;
exports.BatchGetNamedQueryOutput$ = BatchGetNamedQueryOutput$;
exports.BatchGetPreparedStatement$ = BatchGetPreparedStatement$;
exports.BatchGetPreparedStatementCommand = BatchGetPreparedStatementCommand;
exports.BatchGetPreparedStatementInput$ = BatchGetPreparedStatementInput$;
exports.BatchGetPreparedStatementOutput$ = BatchGetPreparedStatementOutput$;
exports.BatchGetQueryExecution$ = BatchGetQueryExecution$;
exports.BatchGetQueryExecutionCommand = BatchGetQueryExecutionCommand;
exports.BatchGetQueryExecutionInput$ = BatchGetQueryExecutionInput$;
exports.BatchGetQueryExecutionOutput$ = BatchGetQueryExecutionOutput$;
exports.CalculationConfiguration$ = CalculationConfiguration$;
exports.CalculationExecutionState = CalculationExecutionState;
exports.CalculationResult$ = CalculationResult$;
exports.CalculationStatistics$ = CalculationStatistics$;
exports.CalculationStatus$ = CalculationStatus$;
exports.CalculationSummary$ = CalculationSummary$;
exports.CancelCapacityReservation$ = CancelCapacityReservation$;
exports.CancelCapacityReservationCommand = CancelCapacityReservationCommand;
exports.CancelCapacityReservationInput$ = CancelCapacityReservationInput$;
exports.CancelCapacityReservationOutput$ = CancelCapacityReservationOutput$;
exports.CapacityAllocation$ = CapacityAllocation$;
exports.CapacityAllocationStatus = CapacityAllocationStatus;
exports.CapacityAssignment$ = CapacityAssignment$;
exports.CapacityAssignmentConfiguration$ = CapacityAssignmentConfiguration$;
exports.CapacityReservation$ = CapacityReservation$;
exports.CapacityReservationStatus = CapacityReservationStatus;
exports.Classification$ = Classification$;
exports.CloudWatchLoggingConfiguration$ = CloudWatchLoggingConfiguration$;
exports.Column$ = Column$;
exports.ColumnInfo$ = ColumnInfo$;
exports.ColumnNullable = ColumnNullable;
exports.ConnectionType = ConnectionType;
exports.CreateCapacityReservation$ = CreateCapacityReservation$;
exports.CreateCapacityReservationCommand = CreateCapacityReservationCommand;
exports.CreateCapacityReservationInput$ = CreateCapacityReservationInput$;
exports.CreateCapacityReservationOutput$ = CreateCapacityReservationOutput$;
exports.CreateDataCatalog$ = CreateDataCatalog$;
exports.CreateDataCatalogCommand = CreateDataCatalogCommand;
exports.CreateDataCatalogInput$ = CreateDataCatalogInput$;
exports.CreateDataCatalogOutput$ = CreateDataCatalogOutput$;
exports.CreateNamedQuery$ = CreateNamedQuery$;
exports.CreateNamedQueryCommand = CreateNamedQueryCommand;
exports.CreateNamedQueryInput$ = CreateNamedQueryInput$;
exports.CreateNamedQueryOutput$ = CreateNamedQueryOutput$;
exports.CreateNotebook$ = CreateNotebook$;
exports.CreateNotebookCommand = CreateNotebookCommand;
exports.CreateNotebookInput$ = CreateNotebookInput$;
exports.CreateNotebookOutput$ = CreateNotebookOutput$;
exports.CreatePreparedStatement$ = CreatePreparedStatement$;
exports.CreatePreparedStatementCommand = CreatePreparedStatementCommand;
exports.CreatePreparedStatementInput$ = CreatePreparedStatementInput$;
exports.CreatePreparedStatementOutput$ = CreatePreparedStatementOutput$;
exports.CreatePresignedNotebookUrl$ = CreatePresignedNotebookUrl$;
exports.CreatePresignedNotebookUrlCommand = CreatePresignedNotebookUrlCommand;
exports.CreatePresignedNotebookUrlRequest$ = CreatePresignedNotebookUrlRequest$;
exports.CreatePresignedNotebookUrlResponse$ = CreatePresignedNotebookUrlResponse$;
exports.CreateWorkGroup$ = CreateWorkGroup$;
exports.CreateWorkGroupCommand = CreateWorkGroupCommand;
exports.CreateWorkGroupInput$ = CreateWorkGroupInput$;
exports.CreateWorkGroupOutput$ = CreateWorkGroupOutput$;
exports.CustomerContentEncryptionConfiguration$ = CustomerContentEncryptionConfiguration$;
exports.DataCatalog$ = DataCatalog$;
exports.DataCatalogStatus = DataCatalogStatus;
exports.DataCatalogSummary$ = DataCatalogSummary$;
exports.DataCatalogType = DataCatalogType;
exports.Database$ = Database$;
exports.Datum$ = Datum$;
exports.DeleteCapacityReservation$ = DeleteCapacityReservation$;
exports.DeleteCapacityReservationCommand = DeleteCapacityReservationCommand;
exports.DeleteCapacityReservationInput$ = DeleteCapacityReservationInput$;
exports.DeleteCapacityReservationOutput$ = DeleteCapacityReservationOutput$;
exports.DeleteDataCatalog$ = DeleteDataCatalog$;
exports.DeleteDataCatalogCommand = DeleteDataCatalogCommand;
exports.DeleteDataCatalogInput$ = DeleteDataCatalogInput$;
exports.DeleteDataCatalogOutput$ = DeleteDataCatalogOutput$;
exports.DeleteNamedQuery$ = DeleteNamedQuery$;
exports.DeleteNamedQueryCommand = DeleteNamedQueryCommand;
exports.DeleteNamedQueryInput$ = DeleteNamedQueryInput$;
exports.DeleteNamedQueryOutput$ = DeleteNamedQueryOutput$;
exports.DeleteNotebook$ = DeleteNotebook$;
exports.DeleteNotebookCommand = DeleteNotebookCommand;
exports.DeleteNotebookInput$ = DeleteNotebookInput$;
exports.DeleteNotebookOutput$ = DeleteNotebookOutput$;
exports.DeletePreparedStatement$ = DeletePreparedStatement$;
exports.DeletePreparedStatementCommand = DeletePreparedStatementCommand;
exports.DeletePreparedStatementInput$ = DeletePreparedStatementInput$;
exports.DeletePreparedStatementOutput$ = DeletePreparedStatementOutput$;
exports.DeleteWorkGroup$ = DeleteWorkGroup$;
exports.DeleteWorkGroupCommand = DeleteWorkGroupCommand;
exports.DeleteWorkGroupInput$ = DeleteWorkGroupInput$;
exports.DeleteWorkGroupOutput$ = DeleteWorkGroupOutput$;
exports.EncryptionConfiguration$ = EncryptionConfiguration$;
exports.EncryptionOption = EncryptionOption;
exports.EngineConfiguration$ = EngineConfiguration$;
exports.EngineVersion$ = EngineVersion$;
exports.ExecutorState = ExecutorState;
exports.ExecutorType = ExecutorType;
exports.ExecutorsSummary$ = ExecutorsSummary$;
exports.ExportNotebook$ = ExportNotebook$;
exports.ExportNotebookCommand = ExportNotebookCommand;
exports.ExportNotebookInput$ = ExportNotebookInput$;
exports.ExportNotebookOutput$ = ExportNotebookOutput$;
exports.FilterDefinition$ = FilterDefinition$;
exports.GetCalculationExecution$ = GetCalculationExecution$;
exports.GetCalculationExecutionCode$ = GetCalculationExecutionCode$;
exports.GetCalculationExecutionCodeCommand = GetCalculationExecutionCodeCommand;
exports.GetCalculationExecutionCodeRequest$ = GetCalculationExecutionCodeRequest$;
exports.GetCalculationExecutionCodeResponse$ = GetCalculationExecutionCodeResponse$;
exports.GetCalculationExecutionCommand = GetCalculationExecutionCommand;
exports.GetCalculationExecutionRequest$ = GetCalculationExecutionRequest$;
exports.GetCalculationExecutionResponse$ = GetCalculationExecutionResponse$;
exports.GetCalculationExecutionStatus$ = GetCalculationExecutionStatus$;
exports.GetCalculationExecutionStatusCommand = GetCalculationExecutionStatusCommand;
exports.GetCalculationExecutionStatusRequest$ = GetCalculationExecutionStatusRequest$;
exports.GetCalculationExecutionStatusResponse$ = GetCalculationExecutionStatusResponse$;
exports.GetCapacityAssignmentConfiguration$ = GetCapacityAssignmentConfiguration$;
exports.GetCapacityAssignmentConfigurationCommand = GetCapacityAssignmentConfigurationCommand;
exports.GetCapacityAssignmentConfigurationInput$ = GetCapacityAssignmentConfigurationInput$;
exports.GetCapacityAssignmentConfigurationOutput$ = GetCapacityAssignmentConfigurationOutput$;
exports.GetCapacityReservation$ = GetCapacityReservation$;
exports.GetCapacityReservationCommand = GetCapacityReservationCommand;
exports.GetCapacityReservationInput$ = GetCapacityReservationInput$;
exports.GetCapacityReservationOutput$ = GetCapacityReservationOutput$;
exports.GetDataCatalog$ = GetDataCatalog$;
exports.GetDataCatalogCommand = GetDataCatalogCommand;
exports.GetDataCatalogInput$ = GetDataCatalogInput$;
exports.GetDataCatalogOutput$ = GetDataCatalogOutput$;
exports.GetDatabase$ = GetDatabase$;
exports.GetDatabaseCommand = GetDatabaseCommand;
exports.GetDatabaseInput$ = GetDatabaseInput$;
exports.GetDatabaseOutput$ = GetDatabaseOutput$;
exports.GetNamedQuery$ = GetNamedQuery$;
exports.GetNamedQueryCommand = GetNamedQueryCommand;
exports.GetNamedQueryInput$ = GetNamedQueryInput$;
exports.GetNamedQueryOutput$ = GetNamedQueryOutput$;
exports.GetNotebookMetadata$ = GetNotebookMetadata$;
exports.GetNotebookMetadataCommand = GetNotebookMetadataCommand;
exports.GetNotebookMetadataInput$ = GetNotebookMetadataInput$;
exports.GetNotebookMetadataOutput$ = GetNotebookMetadataOutput$;
exports.GetPreparedStatement$ = GetPreparedStatement$;
exports.GetPreparedStatementCommand = GetPreparedStatementCommand;
exports.GetPreparedStatementInput$ = GetPreparedStatementInput$;
exports.GetPreparedStatementOutput$ = GetPreparedStatementOutput$;
exports.GetQueryExecution$ = GetQueryExecution$;
exports.GetQueryExecutionCommand = GetQueryExecutionCommand;
exports.GetQueryExecutionInput$ = GetQueryExecutionInput$;
exports.GetQueryExecutionOutput$ = GetQueryExecutionOutput$;
exports.GetQueryResults$ = GetQueryResults$;
exports.GetQueryResultsCommand = GetQueryResultsCommand;
exports.GetQueryResultsInput$ = GetQueryResultsInput$;
exports.GetQueryResultsOutput$ = GetQueryResultsOutput$;
exports.GetQueryRuntimeStatistics$ = GetQueryRuntimeStatistics$;
exports.GetQueryRuntimeStatisticsCommand = GetQueryRuntimeStatisticsCommand;
exports.GetQueryRuntimeStatisticsInput$ = GetQueryRuntimeStatisticsInput$;
exports.GetQueryRuntimeStatisticsOutput$ = GetQueryRuntimeStatisticsOutput$;
exports.GetResourceDashboard$ = GetResourceDashboard$;
exports.GetResourceDashboardCommand = GetResourceDashboardCommand;
exports.GetResourceDashboardRequest$ = GetResourceDashboardRequest$;
exports.GetResourceDashboardResponse$ = GetResourceDashboardResponse$;
exports.GetSession$ = GetSession$;
exports.GetSessionCommand = GetSessionCommand;
exports.GetSessionEndpoint$ = GetSessionEndpoint$;
exports.GetSessionEndpointCommand = GetSessionEndpointCommand;
exports.GetSessionEndpointRequest$ = GetSessionEndpointRequest$;
exports.GetSessionEndpointResponse$ = GetSessionEndpointResponse$;
exports.GetSessionRequest$ = GetSessionRequest$;
exports.GetSessionResponse$ = GetSessionResponse$;
exports.GetSessionStatus$ = GetSessionStatus$;
exports.GetSessionStatusCommand = GetSessionStatusCommand;
exports.GetSessionStatusRequest$ = GetSessionStatusRequest$;
exports.GetSessionStatusResponse$ = GetSessionStatusResponse$;
exports.GetTableMetadata$ = GetTableMetadata$;
exports.GetTableMetadataCommand = GetTableMetadataCommand;
exports.GetTableMetadataInput$ = GetTableMetadataInput$;
exports.GetTableMetadataOutput$ = GetTableMetadataOutput$;
exports.GetWorkGroup$ = GetWorkGroup$;
exports.GetWorkGroupCommand = GetWorkGroupCommand;
exports.GetWorkGroupInput$ = GetWorkGroupInput$;
exports.GetWorkGroupOutput$ = GetWorkGroupOutput$;
exports.IdentityCenterConfiguration$ = IdentityCenterConfiguration$;
exports.ImportNotebook$ = ImportNotebook$;
exports.ImportNotebookCommand = ImportNotebookCommand;
exports.ImportNotebookInput$ = ImportNotebookInput$;
exports.ImportNotebookOutput$ = ImportNotebookOutput$;
exports.InternalServerException = InternalServerException;
exports.InternalServerException$ = InternalServerException$;
exports.InvalidRequestException = InvalidRequestException;
exports.InvalidRequestException$ = InvalidRequestException$;
exports.ListApplicationDPUSizes$ = ListApplicationDPUSizes$;
exports.ListApplicationDPUSizesCommand = ListApplicationDPUSizesCommand;
exports.ListApplicationDPUSizesInput$ = ListApplicationDPUSizesInput$;
exports.ListApplicationDPUSizesOutput$ = ListApplicationDPUSizesOutput$;
exports.ListCalculationExecutions$ = ListCalculationExecutions$;
exports.ListCalculationExecutionsCommand = ListCalculationExecutionsCommand;
exports.ListCalculationExecutionsRequest$ = ListCalculationExecutionsRequest$;
exports.ListCalculationExecutionsResponse$ = ListCalculationExecutionsResponse$;
exports.ListCapacityReservations$ = ListCapacityReservations$;
exports.ListCapacityReservationsCommand = ListCapacityReservationsCommand;
exports.ListCapacityReservationsInput$ = ListCapacityReservationsInput$;
exports.ListCapacityReservationsOutput$ = ListCapacityReservationsOutput$;
exports.ListDataCatalogs$ = ListDataCatalogs$;
exports.ListDataCatalogsCommand = ListDataCatalogsCommand;
exports.ListDataCatalogsInput$ = ListDataCatalogsInput$;
exports.ListDataCatalogsOutput$ = ListDataCatalogsOutput$;
exports.ListDatabases$ = ListDatabases$;
exports.ListDatabasesCommand = ListDatabasesCommand;
exports.ListDatabasesInput$ = ListDatabasesInput$;
exports.ListDatabasesOutput$ = ListDatabasesOutput$;
exports.ListEngineVersions$ = ListEngineVersions$;
exports.ListEngineVersionsCommand = ListEngineVersionsCommand;
exports.ListEngineVersionsInput$ = ListEngineVersionsInput$;
exports.ListEngineVersionsOutput$ = ListEngineVersionsOutput$;
exports.ListExecutors$ = ListExecutors$;
exports.ListExecutorsCommand = ListExecutorsCommand;
exports.ListExecutorsRequest$ = ListExecutorsRequest$;
exports.ListExecutorsResponse$ = ListExecutorsResponse$;
exports.ListNamedQueries$ = ListNamedQueries$;
exports.ListNamedQueriesCommand = ListNamedQueriesCommand;
exports.ListNamedQueriesInput$ = ListNamedQueriesInput$;
exports.ListNamedQueriesOutput$ = ListNamedQueriesOutput$;
exports.ListNotebookMetadata$ = ListNotebookMetadata$;
exports.ListNotebookMetadataCommand = ListNotebookMetadataCommand;
exports.ListNotebookMetadataInput$ = ListNotebookMetadataInput$;
exports.ListNotebookMetadataOutput$ = ListNotebookMetadataOutput$;
exports.ListNotebookSessions$ = ListNotebookSessions$;
exports.ListNotebookSessionsCommand = ListNotebookSessionsCommand;
exports.ListNotebookSessionsRequest$ = ListNotebookSessionsRequest$;
exports.ListNotebookSessionsResponse$ = ListNotebookSessionsResponse$;
exports.ListPreparedStatements$ = ListPreparedStatements$;
exports.ListPreparedStatementsCommand = ListPreparedStatementsCommand;
exports.ListPreparedStatementsInput$ = ListPreparedStatementsInput$;
exports.ListPreparedStatementsOutput$ = ListPreparedStatementsOutput$;
exports.ListQueryExecutions$ = ListQueryExecutions$;
exports.ListQueryExecutionsCommand = ListQueryExecutionsCommand;
exports.ListQueryExecutionsInput$ = ListQueryExecutionsInput$;
exports.ListQueryExecutionsOutput$ = ListQueryExecutionsOutput$;
exports.ListSessions$ = ListSessions$;
exports.ListSessionsCommand = ListSessionsCommand;
exports.ListSessionsRequest$ = ListSessionsRequest$;
exports.ListSessionsResponse$ = ListSessionsResponse$;
exports.ListTableMetadata$ = ListTableMetadata$;
exports.ListTableMetadataCommand = ListTableMetadataCommand;
exports.ListTableMetadataInput$ = ListTableMetadataInput$;
exports.ListTableMetadataOutput$ = ListTableMetadataOutput$;
exports.ListTagsForResource$ = ListTagsForResource$;
exports.ListTagsForResourceCommand = ListTagsForResourceCommand;
exports.ListTagsForResourceInput$ = ListTagsForResourceInput$;
exports.ListTagsForResourceOutput$ = ListTagsForResourceOutput$;
exports.ListWorkGroups$ = ListWorkGroups$;
exports.ListWorkGroupsCommand = ListWorkGroupsCommand;
exports.ListWorkGroupsInput$ = ListWorkGroupsInput$;
exports.ListWorkGroupsOutput$ = ListWorkGroupsOutput$;
exports.ManagedLoggingConfiguration$ = ManagedLoggingConfiguration$;
exports.ManagedQueryResultsConfiguration$ = ManagedQueryResultsConfiguration$;
exports.ManagedQueryResultsConfigurationUpdates$ = ManagedQueryResultsConfigurationUpdates$;
exports.ManagedQueryResultsEncryptionConfiguration$ = ManagedQueryResultsEncryptionConfiguration$;
exports.MetadataException = MetadataException;
exports.MetadataException$ = MetadataException$;
exports.MonitoringConfiguration$ = MonitoringConfiguration$;
exports.NamedQuery$ = NamedQuery$;
exports.NotebookMetadata$ = NotebookMetadata$;
exports.NotebookSessionSummary$ = NotebookSessionSummary$;
exports.NotebookType = NotebookType;
exports.PreparedStatement$ = PreparedStatement$;
exports.PreparedStatementSummary$ = PreparedStatementSummary$;
exports.PutCapacityAssignmentConfiguration$ = PutCapacityAssignmentConfiguration$;
exports.PutCapacityAssignmentConfigurationCommand = PutCapacityAssignmentConfigurationCommand;
exports.PutCapacityAssignmentConfigurationInput$ = PutCapacityAssignmentConfigurationInput$;
exports.PutCapacityAssignmentConfigurationOutput$ = PutCapacityAssignmentConfigurationOutput$;
exports.QueryExecution$ = QueryExecution$;
exports.QueryExecutionContext$ = QueryExecutionContext$;
exports.QueryExecutionState = QueryExecutionState;
exports.QueryExecutionStatistics$ = QueryExecutionStatistics$;
exports.QueryExecutionStatus$ = QueryExecutionStatus$;
exports.QueryResultType = QueryResultType;
exports.QueryResultsS3AccessGrantsConfiguration$ = QueryResultsS3AccessGrantsConfiguration$;
exports.QueryRuntimeStatistics$ = QueryRuntimeStatistics$;
exports.QueryRuntimeStatisticsRows$ = QueryRuntimeStatisticsRows$;
exports.QueryRuntimeStatisticsTimeline$ = QueryRuntimeStatisticsTimeline$;
exports.QueryStage$ = QueryStage$;
exports.QueryStagePlanNode$ = QueryStagePlanNode$;
exports.ResourceNotFoundException = ResourceNotFoundException;
exports.ResourceNotFoundException$ = ResourceNotFoundException$;
exports.ResultConfiguration$ = ResultConfiguration$;
exports.ResultConfigurationUpdates$ = ResultConfigurationUpdates$;
exports.ResultReuseByAgeConfiguration$ = ResultReuseByAgeConfiguration$;
exports.ResultReuseConfiguration$ = ResultReuseConfiguration$;
exports.ResultReuseInformation$ = ResultReuseInformation$;
exports.ResultSet$ = ResultSet$;
exports.ResultSetMetadata$ = ResultSetMetadata$;
exports.Row$ = Row$;
exports.S3AclOption = S3AclOption;
exports.S3LoggingConfiguration$ = S3LoggingConfiguration$;
exports.SessionAlreadyExistsException = SessionAlreadyExistsException;
exports.SessionAlreadyExistsException$ = SessionAlreadyExistsException$;
exports.SessionConfiguration$ = SessionConfiguration$;
exports.SessionState = SessionState;
exports.SessionStatistics$ = SessionStatistics$;
exports.SessionStatus$ = SessionStatus$;
exports.SessionSummary$ = SessionSummary$;
exports.StartCalculationExecution$ = StartCalculationExecution$;
exports.StartCalculationExecutionCommand = StartCalculationExecutionCommand;
exports.StartCalculationExecutionRequest$ = StartCalculationExecutionRequest$;
exports.StartCalculationExecutionResponse$ = StartCalculationExecutionResponse$;
exports.StartQueryExecution$ = StartQueryExecution$;
exports.StartQueryExecutionCommand = StartQueryExecutionCommand;
exports.StartQueryExecutionInput$ = StartQueryExecutionInput$;
exports.StartQueryExecutionOutput$ = StartQueryExecutionOutput$;
exports.StartSession$ = StartSession$;
exports.StartSessionCommand = StartSessionCommand;
exports.StartSessionRequest$ = StartSessionRequest$;
exports.StartSessionResponse$ = StartSessionResponse$;
exports.StatementType = StatementType;
exports.StopCalculationExecution$ = StopCalculationExecution$;
exports.StopCalculationExecutionCommand = StopCalculationExecutionCommand;
exports.StopCalculationExecutionRequest$ = StopCalculationExecutionRequest$;
exports.StopCalculationExecutionResponse$ = StopCalculationExecutionResponse$;
exports.StopQueryExecution$ = StopQueryExecution$;
exports.StopQueryExecutionCommand = StopQueryExecutionCommand;
exports.StopQueryExecutionInput$ = StopQueryExecutionInput$;
exports.StopQueryExecutionOutput$ = StopQueryExecutionOutput$;
exports.TableMetadata$ = TableMetadata$;
exports.Tag$ = Tag$;
exports.TagResource$ = TagResource$;
exports.TagResourceCommand = TagResourceCommand;
exports.TagResourceInput$ = TagResourceInput$;
exports.TagResourceOutput$ = TagResourceOutput$;
exports.TerminateSession$ = TerminateSession$;
exports.TerminateSessionCommand = TerminateSessionCommand;
exports.TerminateSessionRequest$ = TerminateSessionRequest$;
exports.TerminateSessionResponse$ = TerminateSessionResponse$;
exports.ThrottleReason = ThrottleReason;
exports.TooManyRequestsException = TooManyRequestsException;
exports.TooManyRequestsException$ = TooManyRequestsException$;
exports.UnprocessedNamedQueryId$ = UnprocessedNamedQueryId$;
exports.UnprocessedPreparedStatementName$ = UnprocessedPreparedStatementName$;
exports.UnprocessedQueryExecutionId$ = UnprocessedQueryExecutionId$;
exports.UntagResource$ = UntagResource$;
exports.UntagResourceCommand = UntagResourceCommand;
exports.UntagResourceInput$ = UntagResourceInput$;
exports.UntagResourceOutput$ = UntagResourceOutput$;
exports.UpdateCapacityReservation$ = UpdateCapacityReservation$;
exports.UpdateCapacityReservationCommand = UpdateCapacityReservationCommand;
exports.UpdateCapacityReservationInput$ = UpdateCapacityReservationInput$;
exports.UpdateCapacityReservationOutput$ = UpdateCapacityReservationOutput$;
exports.UpdateDataCatalog$ = UpdateDataCatalog$;
exports.UpdateDataCatalogCommand = UpdateDataCatalogCommand;
exports.UpdateDataCatalogInput$ = UpdateDataCatalogInput$;
exports.UpdateDataCatalogOutput$ = UpdateDataCatalogOutput$;
exports.UpdateNamedQuery$ = UpdateNamedQuery$;
exports.UpdateNamedQueryCommand = UpdateNamedQueryCommand;
exports.UpdateNamedQueryInput$ = UpdateNamedQueryInput$;
exports.UpdateNamedQueryOutput$ = UpdateNamedQueryOutput$;
exports.UpdateNotebook$ = UpdateNotebook$;
exports.UpdateNotebookCommand = UpdateNotebookCommand;
exports.UpdateNotebookInput$ = UpdateNotebookInput$;
exports.UpdateNotebookMetadata$ = UpdateNotebookMetadata$;
exports.UpdateNotebookMetadataCommand = UpdateNotebookMetadataCommand;
exports.UpdateNotebookMetadataInput$ = UpdateNotebookMetadataInput$;
exports.UpdateNotebookMetadataOutput$ = UpdateNotebookMetadataOutput$;
exports.UpdateNotebookOutput$ = UpdateNotebookOutput$;
exports.UpdatePreparedStatement$ = UpdatePreparedStatement$;
exports.UpdatePreparedStatementCommand = UpdatePreparedStatementCommand;
exports.UpdatePreparedStatementInput$ = UpdatePreparedStatementInput$;
exports.UpdatePreparedStatementOutput$ = UpdatePreparedStatementOutput$;
exports.UpdateWorkGroup$ = UpdateWorkGroup$;
exports.UpdateWorkGroupCommand = UpdateWorkGroupCommand;
exports.UpdateWorkGroupInput$ = UpdateWorkGroupInput$;
exports.UpdateWorkGroupOutput$ = UpdateWorkGroupOutput$;
exports.WorkGroup$ = WorkGroup$;
exports.WorkGroupConfiguration$ = WorkGroupConfiguration$;
exports.WorkGroupConfigurationUpdates$ = WorkGroupConfigurationUpdates$;
exports.WorkGroupState = WorkGroupState;
exports.WorkGroupSummary$ = WorkGroupSummary$;
exports.paginateGetQueryResults = paginateGetQueryResults;
exports.paginateListApplicationDPUSizes = paginateListApplicationDPUSizes;
exports.paginateListCalculationExecutions = paginateListCalculationExecutions;
exports.paginateListCapacityReservations = paginateListCapacityReservations;
exports.paginateListDataCatalogs = paginateListDataCatalogs;
exports.paginateListDatabases = paginateListDatabases;
exports.paginateListEngineVersions = paginateListEngineVersions;
exports.paginateListExecutors = paginateListExecutors;
exports.paginateListNamedQueries = paginateListNamedQueries;
exports.paginateListPreparedStatements = paginateListPreparedStatements;
exports.paginateListQueryExecutions = paginateListQueryExecutions;
exports.paginateListSessions = paginateListSessions;
exports.paginateListTableMetadata = paginateListTableMetadata;
exports.paginateListTagsForResource = paginateListTagsForResource;
exports.paginateListWorkGroups = paginateListWorkGroups;
